[*] Start mutation process:
   - targets: graph1.py
   - tests: ts-3-5/test_3-5_1-0_309.py
[*] 12 tests passed:
   - test_3-5_1-0_309 [0.18356 s]
[*] Start mutants generation and execution:
   - [#   1] AOD graph1: 
--------------------------------------------------------------------------------
   57:         visited = [0] * self.verticies
   58:         
   59:         def valid(node):
   60:             
-  61:             if visited[node] == -1:
+  61:             if visited[node] == 1:
   62:                 return False
   63:             elif visited[node] == 1:
   64:                 return True
   65:             visited[node] = -1
--------------------------------------------------------------------------------
[0.21567 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_has_cycle

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad5a8a620>

    def test_has_cycle(self):
        graph = Graph(3)
        graph.add_edge(0, 1)
        graph.add_edge(1, 2)
        assert not graph.has_cycle()
    
        graph.add_edge(2, 0)
>       assert graph.has_cycle()

ts-3-5/test_3-5_1-0_309.py:49: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
graph1:77: in has_cycle
    ???
graph1:67: in valid
    ???
graph1:67: in valid
    ???
graph1:67: in valid
    ???
graph1:67: in valid
    ???
E   RecursionError: maximum recursion depth exceeded in comparison
!!! Recursion detected (same locals & position)
   - [#   2] AOD graph1: 
--------------------------------------------------------------------------------
   61:             if visited[node] == -1:
   62:                 return False
   63:             elif visited[node] == 1:
   64:                 return True
-  65:             visited[node] = -1
+  65:             visited[node] = 1
   66:             for neighbor in self.graph[node]:
   67:                 if not (valid(neighbor)):
   68:                     return False
   69:             return True
--------------------------------------------------------------------------------
[0.20411 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_has_cycle

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad5a25fc0>

    def test_has_cycle(self):
        graph = Graph(3)
        graph.add_edge(0, 1)
        graph.add_edge(1, 2)
        assert not graph.has_cycle()
    
        graph.add_edge(2, 0)
>       assert graph.has_cycle()
E       AssertionError

ts-3-5/test_3-5_1-0_309.py:49: AssertionError
   - [#   3] AOR graph1: 
--------------------------------------------------------------------------------
   53:             False if the graph is acyclic.
   54: 
   55:         '''
   56:         
-  57:         visited = [0] * self.verticies
+  57:         visited = [0] / self.verticies
   58:         
   59:         def valid(node):
   60:             
   61:             if visited[node] == -1:
--------------------------------------------------------------------------------
[0.20963 s] incompetent

TypeError: graph1:57: TypeError: unsupported operand type(s) for /: 'list' and 'int'

   - [#   4] AOR graph1: 
--------------------------------------------------------------------------------
   53:             False if the graph is acyclic.
   54: 
   55:         '''
   56:         
-  57:         visited = [0] * self.verticies
+  57:         visited = [0] // self.verticies
   58:         
   59:         def valid(node):
   60:             
   61:             if visited[node] == -1:
--------------------------------------------------------------------------------
[0.22468 s] incompetent

TypeError: graph1:57: TypeError: unsupported operand type(s) for //: 'list' and 'int'

   - [#   5] AOR graph1: 
--------------------------------------------------------------------------------
   53:             False if the graph is acyclic.
   54: 
   55:         '''
   56:         
-  57:         visited = [0] * self.verticies
+  57:         visited = [0] ** self.verticies
   58:         
   59:         def valid(node):
   60:             
   61:             if visited[node] == -1:
--------------------------------------------------------------------------------
[0.20669 s] incompetent

TypeError: graph1:57: TypeError: unsupported operand type(s) for ** or pow(): 'list' and 'int'

   - [#   6] AOR graph1: 
--------------------------------------------------------------------------------
   57:         visited = [0] * self.verticies
   58:         
   59:         def valid(node):
   60:             
-  61:             if visited[node] == -1:
+  61:             if visited[node] == +1:
   62:                 return False
   63:             elif visited[node] == 1:
   64:                 return True
   65:             visited[node] = -1
--------------------------------------------------------------------------------
[0.21588 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_has_cycle

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad560f3d0>

    def test_has_cycle(self):
        graph = Graph(3)
        graph.add_edge(0, 1)
        graph.add_edge(1, 2)
        assert not graph.has_cycle()
    
        graph.add_edge(2, 0)
>       assert graph.has_cycle()

ts-3-5/test_3-5_1-0_309.py:49: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
graph1:77: in has_cycle
    ???
graph1:67: in valid
    ???
graph1:67: in valid
    ???
graph1:67: in valid
    ???
graph1:67: in valid
    ???
E   RecursionError: maximum recursion depth exceeded in comparison
!!! Recursion detected (same locals & position)
   - [#   7] AOR graph1: 
--------------------------------------------------------------------------------
   61:             if visited[node] == -1:
   62:                 return False
   63:             elif visited[node] == 1:
   64:                 return True
-  65:             visited[node] = -1
+  65:             visited[node] = +1
   66:             for neighbor in self.graph[node]:
   67:                 if not (valid(neighbor)):
   68:                     return False
   69:             return True
--------------------------------------------------------------------------------
[0.20936 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_has_cycle

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad586b220>

    def test_has_cycle(self):
        graph = Graph(3)
        graph.add_edge(0, 1)
        graph.add_edge(1, 2)
        assert not graph.has_cycle()
    
        graph.add_edge(2, 0)
>       assert graph.has_cycle()
E       AssertionError

ts-3-5/test_3-5_1-0_309.py:49: AssertionError
   - [#   8] AOR graph1: 
--------------------------------------------------------------------------------
  127:         self.val = val
  128:         self.adjacent = {}
  129:     
  130:     def __str__(self):
- 131:         return (str(self.val) + ' adjacent: ') + str([x for x in self.adjacent])
+ 131:         return (str(self.val) - ' adjacent: ') + str([x for x in self.adjacent])
  132:     
  133:     def add_adjacent(self, node, weight=0):
  134:         self.adjacent[node] = weight
  135:     
--------------------------------------------------------------------------------
[0.18225 s] survived
   - [#   9] AOR graph1: 
--------------------------------------------------------------------------------
  127:         self.val = val
  128:         self.adjacent = {}
  129:     
  130:     def __str__(self):
- 131:         return (str(self.val) + ' adjacent: ') + str([x for x in self.adjacent])
+ 131:         return (str(self.val) + ' adjacent: ') - str([x for x in self.adjacent])
  132:     
  133:     def add_adjacent(self, node, weight=0):
  134:         self.adjacent[node] = weight
  135:     
--------------------------------------------------------------------------------
[0.18380 s] survived
   - [#  10] CDI graph1: 
--------------------------------------------------------------------------------
    1: from collections import defaultdict, deque
    2: 
    3: class GraphNode:
    4:     
-   5:     def __init__(self, val):
-   6:         self.val = val
-   7:         self.adjacent_list = set()
-   8:     
+   5:     @classmethod
+   6:     def __init__(self, val):
+   7:         self.val = val
+   8:         self.adjacent_list = set()
    9:     def add_adjacent(self, node):
   10:         self.adjacent_list.add(node)
   11:     
   12:     def remove_adjacent(self, node):
--------------------------------------------------------------------------------
[0.18742 s] survived
   - [#  11] CDI graph1: 
--------------------------------------------------------------------------------
    5:     def __init__(self, val):
    6:         self.val = val
    7:         self.adjacent_list = set()
    8:     
-   9:     def add_adjacent(self, node):
-  10:         self.adjacent_list.add(node)
-  11:     
+   9:     @classmethod
+  10:     def add_adjacent(self, node):
+  11:         self.adjacent_list.add(node)
   12:     def remove_adjacent(self, node):
   13:         self.adjacent_list.remove(node)
   14: 
   15: class Graph:
--------------------------------------------------------------------------------
[0.20346 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraphNode::test_add_adjacent

self = <test_3-5_1-0_309.TestGraphNode object at 0x7d3ad57e6110>

    def test_add_adjacent(self):
        node1 = GraphNode(1)
        node2 = GraphNode(2)
>       node1.add_adjacent(node2)

ts-3-5/test_3-5_1-0_309.py:23: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <class 'graph1.GraphNode'>
node = <graph1.GraphNode object at 0x7d3ad57e52d0>

>   ???
E   AttributeError: type object 'GraphNode' has no attribute 'adjacent_list'

graph1:11: AttributeError
   - [#  12] CDI graph1: 
--------------------------------------------------------------------------------
    8:     
    9:     def add_adjacent(self, node):
   10:         self.adjacent_list.add(node)
   11:     
-  12:     def remove_adjacent(self, node):
-  13:         self.adjacent_list.remove(node)
-  14: 
+  12:     @classmethod
+  13:     def remove_adjacent(self, node):
+  14:         self.adjacent_list.remove(node)
   15: class Graph:
   16:     '''
   17:     A directed graph represented with an adjacency list.
   18:     '''
--------------------------------------------------------------------------------
[0.18942 s] survived
   - [#  13] CDI graph1: 
--------------------------------------------------------------------------------
   16:     '''
   17:     A directed graph represented with an adjacency list.
   18:     '''
   19:     
-  20:     def __init__(self, verticies):
-  21:         self.graph = {}
-  22:         self.verticies = verticies
-  23:     
+  20:     @classmethod
+  21:     def __init__(self, verticies):
+  22:         self.graph = {}
+  23:         self.verticies = verticies
   24:     def add_edge(self, source, destination):
   25:         '''
   26:         Add an edge to the graph.
   27: 
--------------------------------------------------------------------------------
[0.18214 s] survived
   - [#  14] CDI graph1: 
--------------------------------------------------------------------------------
   20:     def __init__(self, verticies):
   21:         self.graph = {}
   22:         self.verticies = verticies
   23:     
-  24:     def add_edge(self, source, destination):
-  25:         '''
-  26:         Add an edge to the graph.
-  27: 
-  28:         Add an edge pointing from source vertex
-  29:         to destination vertex.
-  30: 
-  31:         Args:
-  32:             source: the source vertex
-  33:             destination: the destination vertex
-  34: 
-  35:         '''
-  36:         if len(self.graph) > self.verticies:
-  37:             raise IndexError('Too many verticies in graph.')
-  38:         
-  39:         if source in self.graph:
-  40:             self.graph[source].append(destination)
-  41:         else:
-  42:             self.graph[source] = [destination]
-  43:         
-  44:         if destination not in self.graph:
-  45:             self.graph[destination] = []
-  46:     
+  24:     @classmethod
+  25:     def add_edge(self, source, destination):
+  26:         '''
+  27:         Add an edge to the graph.
+  28: 
+  29:         Add an edge pointing from source vertex
+  30:         to destination vertex.
+  31: 
+  32:         Args:
+  33:             source: the source vertex
+  34:             destination: the destination vertex
+  35: 
+  36:         '''
+  37:         if len(self.graph) > self.verticies:
+  38:             raise IndexError('Too many verticies in graph.')
+  39:         
+  40:         if source in self.graph:
+  41:             self.graph[source].append(destination)
+  42:         else:
+  43:             self.graph[source] = [destination]
+  44:         
+  45:         if destination not in self.graph:
+  46:             self.graph[destination] = []
   47:     def has_cycle(self):
   48:         '''
   49:         Detect if a graph has a cycle.
   50: 
--------------------------------------------------------------------------------
[0.20950 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_add_edge

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad5851c00>

    def test_add_edge(self):
        graph = Graph(5)
>       graph.add_edge(1, 2)

ts-3-5/test_3-5_1-0_309.py:37: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <class 'graph1.Graph'>, source = 1, destination = 2

>   ???
E   AttributeError: type object 'Graph' has no attribute 'graph'

graph1:37: AttributeError
   - [#  15] CDI graph1: 
--------------------------------------------------------------------------------
   43:         
   44:         if destination not in self.graph:
   45:             self.graph[destination] = []
   46:     
-  47:     def has_cycle(self):
-  48:         '''
-  49:         Detect if a graph has a cycle.
-  50: 
-  51:         Returns:
-  52:             True if the graph has a cycle and
-  53:             False if the graph is acyclic.
-  54: 
-  55:         '''
-  56:         
-  57:         visited = [0] * self.verticies
-  58:         
-  59:         def valid(node):
-  60:             
-  61:             if visited[node] == -1:
-  62:                 return False
-  63:             elif visited[node] == 1:
-  64:                 return True
-  65:             visited[node] = -1
-  66:             for neighbor in self.graph[node]:
-  67:                 if not (valid(neighbor)):
-  68:                     return False
-  69:             return True
-  70:         
-  71:         for node in range(self.verticies):
-  72:             
+  47:     @classmethod
+  48:     def has_cycle(self):
+  49:         '''
+  50:         Detect if a graph has a cycle.
+  51: 
+  52:         Returns:
+  53:             True if the graph has a cycle and
+  54:             False if the graph is acyclic.
+  55: 
+  56:         '''
+  57:         
+  58:         visited = [0] * self.verticies
+  59:         
+  60:         def valid(node):
+  61:             
+  62:             if visited[node] == -1:
+  63:                 return False
+  64:             elif visited[node] == 1:
+  65:                 return True
+  66:             visited[node] = -1
+  67:             for neighbor in self.graph[node]:
+  68:                 if not (valid(neighbor)):
+  69:                     return False
+  70:             return True
+  71:         
+  72:         for node in range(self.verticies):
   73:             
   74:             
-  75:             if not (visited[node]):
-  76:                 
-  77:                 if valid(node):
-  78:                     
-  79:                     visited = list(map((lambda x: (abs(x))), visited))
-  80:                 else:
-  81:                     
-  82:                     return True
-  83:         
-  84:         return False
-  85:     
+  75:             
+  76:             if not (visited[node]):
+  77:                 
+  78:                 if valid(node):
+  79:                     
+  80:                     visited = list(map((lambda x: (abs(x))), visited))
+  81:                 else:
+  82:                     
+  83:                     return True
+  84:         
+  85:         return False
   86:     def topological_sort(self):
   87:         """
   88:         Sort the graph topologically.
   89: 
--------------------------------------------------------------------------------
[0.20568 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_has_cycle

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad56e9d80>

    def test_has_cycle(self):
        graph = Graph(3)
        graph.add_edge(0, 1)
        graph.add_edge(1, 2)
>       assert not graph.has_cycle()

ts-3-5/test_3-5_1-0_309.py:46: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <class 'graph1.Graph'>

>   ???
E   AttributeError: type object 'Graph' has no attribute 'verticies'

graph1:58: AttributeError
   - [#  16] CDI graph1: 
--------------------------------------------------------------------------------
   82:                     return True
   83:         
   84:         return False
   85:     
-  86:     def topological_sort(self):
-  87:         """
-  88:         Sort the graph topologically.
-  89: 
-  90:         A topological sort lists nodes in such a way
-  91:         that every node 's' in 's' -> 'd' directed pairs
-  92:         is listed before 'd.'  This will not work in a 
-  93:         graph that contains cycles.
-  94: 
-  95:         The algorithm looks at every node, and does a
-  96:         dfs for each node adjacent to the node and then adds
-  97:         the originating node to a stack once all adjacent
-  98:         nodes have been searched.  In the end, the stack
-  99:         will be in order of a possible topological sort.
- 100: 
- 101:         Topological sorts are not necessarily unique.
- 102: 
- 103:         Returns:
- 104:             A list of vertices in a topological ordering.
- 105: 
- 106:         """
- 107:         visited = set()
- 108:         stack = deque()
- 109:         
- 110:         def dfs(vertex):
- 111:             visited.add(vertex)
- 112:             for j in self.graph[vertex]:
- 113:                 if j not in visited:
- 114:                     dfs(j)
- 115:             
- 116:             stack.appendleft(vertex)
- 117:         
- 118:         for (key, _) in self.graph.items():
- 119:             if key not in visited:
- 120:                 dfs(key)
- 121:         
- 122:         return list(stack)
- 123: 
+  86:     @classmethod
+  87:     def topological_sort(self):
+  88:         """
+  89:         Sort the graph topologically.
+  90: 
+  91:         A topological sort lists nodes in such a way
+  92:         that every node 's' in 's' -> 'd' directed pairs
+  93:         is listed before 'd.'  This will not work in a 
+  94:         graph that contains cycles.
+  95: 
+  96:         The algorithm looks at every node, and does a
+  97:         dfs for each node adjacent to the node and then adds
+  98:         the originating node to a stack once all adjacent
+  99:         nodes have been searched.  In the end, the stack
+ 100:         will be in order of a possible topological sort.
+ 101: 
+ 102:         Topological sorts are not necessarily unique.
+ 103: 
+ 104:         Returns:
+ 105:             A list of vertices in a topological ordering.
+ 106: 
+ 107:         """
+ 108:         visited = set()
+ 109:         stack = deque()
+ 110:         
+ 111:         def dfs(vertex):
+ 112:             visited.add(vertex)
+ 113:             for j in self.graph[vertex]:
+ 114:                 if j not in visited:
+ 115:                     dfs(j)
+ 116:             
+ 117:             stack.appendleft(vertex)
+ 118:         
+ 119:         for (key, _) in self.graph.items():
+ 120:             if key not in visited:
+ 121:                 dfs(key)
+ 122:         
+ 123:         return list(stack)
  124: class WeightedGraphNode:
  125:     
  126:     def __init__(self, val):
  127:         self.val = val
--------------------------------------------------------------------------------
[0.22962 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_topological_sort

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad55ecfa0>

    def test_topological_sort(self):
        graph = Graph(4)
        graph.add_edge(0, 1)
        graph.add_edge(1, 2)
        graph.add_edge(1, 3)
        graph.add_edge(2, 3)
>       assert graph.topological_sort() == [0, 1, 2, 3]

ts-3-5/test_3-5_1-0_309.py:57: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <class 'graph1.Graph'>

>   ???
E   AttributeError: type object 'Graph' has no attribute 'graph'

graph1:119: AttributeError
   - [#  17] CDI graph1: 
--------------------------------------------------------------------------------
  122:         return list(stack)
  123: 
  124: class WeightedGraphNode:
  125:     
- 126:     def __init__(self, val):
- 127:         self.val = val
- 128:         self.adjacent = {}
- 129:     
+ 126:     @classmethod
+ 127:     def __init__(self, val):
+ 128:         self.val = val
+ 129:         self.adjacent = {}
  130:     def __str__(self):
  131:         return (str(self.val) + ' adjacent: ') + str([x for x in self.adjacent])
  132:     
  133:     def add_adjacent(self, node, weight=0):
--------------------------------------------------------------------------------
[0.18866 s] survived
   - [#  18] CDI graph1: 
--------------------------------------------------------------------------------
  126:     def __init__(self, val):
  127:         self.val = val
  128:         self.adjacent = {}
  129:     
- 130:     def __str__(self):
- 131:         return (str(self.val) + ' adjacent: ') + str([x for x in self.adjacent])
- 132:     
+ 130:     @classmethod
+ 131:     def __str__(self):
+ 132:         return (str(self.val) + ' adjacent: ') + str([x for x in self.adjacent])
  133:     def add_adjacent(self, node, weight=0):
  134:         self.adjacent[node] = weight
  135:     
  136:     def remove_adjacent(self, node):
--------------------------------------------------------------------------------
[0.16927 s] survived
   - [#  19] CDI graph1: 
--------------------------------------------------------------------------------
  129:     
  130:     def __str__(self):
  131:         return (str(self.val) + ' adjacent: ') + str([x for x in self.adjacent])
  132:     
- 133:     def add_adjacent(self, node, weight=0):
- 134:         self.adjacent[node] = weight
- 135:     
+ 133:     @classmethod
+ 134:     def add_adjacent(self, node, weight=0):
+ 135:         self.adjacent[node] = weight
  136:     def remove_adjacent(self, node):
  137:         del self.adjacent[node]
  138: 
  139: class WeightedGraph:
--------------------------------------------------------------------------------
[0.20794 s] killed by ts-3-5/test_3-5_1-0_309.py::TestWeightedGraphNode::test_add_adjacent

self = <test_3-5_1-0_309.TestWeightedGraphNode object at 0x7d3ad572e200>

    def test_add_adjacent(self):
        w_node1 = WeightedGraphNode(10)
        w_node2 = WeightedGraphNode(20)
>       w_node1.add_adjacent(w_node2, 5)

ts-3-5/test_3-5_1-0_309.py:71: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <class 'graph1.WeightedGraphNode'>
node = <graph1.WeightedGraphNode object at 0x7d3ad572cd30>, weight = 5

>   ???
E   AttributeError: type object 'WeightedGraphNode' has no attribute 'adjacent'

graph1:135: AttributeError
   - [#  20] CDI graph1: 
--------------------------------------------------------------------------------
  132:     
  133:     def add_adjacent(self, node, weight=0):
  134:         self.adjacent[node] = weight
  135:     
- 136:     def remove_adjacent(self, node):
- 137:         del self.adjacent[node]
- 138: 
+ 136:     @classmethod
+ 137:     def remove_adjacent(self, node):
+ 138:         del self.adjacent[node]
  139: class WeightedGraph:
  140:     
  141:     def __init__(self):
  142:         self.vertices = {}
--------------------------------------------------------------------------------
[0.21227 s] killed by ts-3-5/test_3-5_1-0_309.py::TestWeightedGraphNode::test_remove_adjacent

self = <test_3-5_1-0_309.TestWeightedGraphNode object at 0x7d3ad54e7d00>

    def test_remove_adjacent(self):
        w_node1 = WeightedGraphNode(10)
        w_node2 = WeightedGraphNode(20)
        w_node1.add_adjacent(w_node2, 5)
>       w_node1.remove_adjacent(w_node2)

ts-3-5/test_3-5_1-0_309.py:79: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <class 'graph1.WeightedGraphNode'>
node = <graph1.WeightedGraphNode object at 0x7d3ad54e4190>

>   ???
E   AttributeError: type object 'WeightedGraphNode' has no attribute 'adjacent'

graph1:138: AttributeError
   - [#  21] CDI graph1: 
--------------------------------------------------------------------------------
  137:         del self.adjacent[node]
  138: 
  139: class WeightedGraph:
  140:     
- 141:     def __init__(self):
- 142:         self.vertices = {}
- 143:     
+ 141:     @classmethod
+ 142:     def __init__(self):
+ 143:         self.vertices = {}
  144:     def __str__(self):
  145:         return ', '.join([node.__str__() for node in self.vertices.values()])
  146:     
  147:     def add_vertex(self, val):
--------------------------------------------------------------------------------
[0.18762 s] survived
   - [#  22] CDI graph1: 
--------------------------------------------------------------------------------
  140:     
  141:     def __init__(self):
  142:         self.vertices = {}
  143:     
- 144:     def __str__(self):
- 145:         return ', '.join([node.__str__() for node in self.vertices.values()])
- 146:     
+ 144:     @classmethod
+ 145:     def __str__(self):
+ 146:         return ', '.join([node.__str__() for node in self.vertices.values()])
  147:     def add_vertex(self, val):
  148:         vertex = WeightedGraphNode(val)
  149:         self.vertices[val] = vertex
  150:     
--------------------------------------------------------------------------------
[0.18567 s] survived
   - [#  23] CDI graph1: 
--------------------------------------------------------------------------------
  143:     
  144:     def __str__(self):
  145:         return ', '.join([node.__str__() for node in self.vertices.values()])
  146:     
- 147:     def add_vertex(self, val):
- 148:         vertex = WeightedGraphNode(val)
- 149:         self.vertices[val] = vertex
- 150:     
+ 147:     @classmethod
+ 148:     def add_vertex(self, val):
+ 149:         vertex = WeightedGraphNode(val)
+ 150:         self.vertices[val] = vertex
  151:     def add_edge(self, src, dst, weight=0):
  152:         if src not in self.vertices:
  153:             self.add_vertex(src)
  154:         if dst not in self.vertices:
--------------------------------------------------------------------------------
[0.21463 s] killed by ts-3-5/test_3-5_1-0_309.py::TestWeightedGraph::test_add_edge

self = <test_3-5_1-0_309.TestWeightedGraph object at 0x7d3ad575e590>

    def test_add_edge(self):
        w_graph = WeightedGraph()
>       w_graph.add_edge(1, 2, 5)

ts-3-5/test_3-5_1-0_309.py:92: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
graph1:153: in add_edge
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <class 'graph1.WeightedGraph'>, val = 1

>   ???
E   AttributeError: type object 'WeightedGraph' has no attribute 'vertices'

graph1:150: AttributeError
   - [#  24] CDI graph1: 
--------------------------------------------------------------------------------
  147:     def add_vertex(self, val):
  148:         vertex = WeightedGraphNode(val)
  149:         self.vertices[val] = vertex
  150:     
- 151:     def add_edge(self, src, dst, weight=0):
- 152:         if src not in self.vertices:
- 153:             self.add_vertex(src)
- 154:         if dst not in self.vertices:
- 155:             self.add_vertex(dst)
- 156:         self.vertices[src].add_adjacent(dst, weight)
- 157:     
+ 151:     @classmethod
+ 152:     def add_edge(self, src, dst, weight=0):
+ 153:         if src not in self.vertices:
+ 154:             self.add_vertex(src)
+ 155:         if dst not in self.vertices:
+ 156:             self.add_vertex(dst)
+ 157:         self.vertices[src].add_adjacent(dst, weight)
  158:     def remove_edge(self, src, dst):
  159:         if src not in self.vertices:
  160:             return
  161:         if dst not in self.vertices:
--------------------------------------------------------------------------------
[0.19317 s] killed by ts-3-5/test_3-5_1-0_309.py::TestWeightedGraph::test_add_edge

self = <test_3-5_1-0_309.TestWeightedGraph object at 0x7d3ad562d390>

    def test_add_edge(self):
        w_graph = WeightedGraph()
>       w_graph.add_edge(1, 2, 5)

ts-3-5/test_3-5_1-0_309.py:92: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <class 'graph1.WeightedGraph'>, src = 1, dst = 2, weight = 5

>   ???
E   AttributeError: type object 'WeightedGraph' has no attribute 'vertices'

graph1:153: AttributeError
   - [#  25] CDI graph1: 
--------------------------------------------------------------------------------
  154:         if dst not in self.vertices:
  155:             self.add_vertex(dst)
  156:         self.vertices[src].add_adjacent(dst, weight)
  157:     
- 158:     def remove_edge(self, src, dst):
- 159:         if src not in self.vertices:
- 160:             return
- 161:         if dst not in self.vertices:
- 162:             return
- 163:         if dst in self.vertices[src].adjacent:
- 164:             self.vertices[src].remove_adjacent(dst)
+ 158:     @classmethod
+ 159:     def remove_edge(self, src, dst):
+ 160:         if src not in self.vertices:
+ 161:             return
+ 162:         if dst not in self.vertices:
+ 163:             return
+ 164:         if dst in self.vertices[src].adjacent:
+ 165:             self.vertices[src].remove_adjacent(dst)
--------------------------------------------------------------------------------
[0.21254 s] killed by ts-3-5/test_3-5_1-0_309.py::TestWeightedGraph::test_remove_edge

self = <test_3-5_1-0_309.TestWeightedGraph object at 0x7d3ad55a88b0>

    def test_remove_edge(self):
        w_graph = WeightedGraph()
        w_graph.add_edge(1, 2, 5)
>       w_graph.remove_edge(1, 2)

ts-3-5/test_3-5_1-0_309.py:100: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <class 'graph1.WeightedGraph'>, src = 1, dst = 2

>   ???
E   AttributeError: type object 'WeightedGraph' has no attribute 'vertices'

graph1:160: AttributeError
   - [#  26] COD graph1: 
--------------------------------------------------------------------------------
   40:             self.graph[source].append(destination)
   41:         else:
   42:             self.graph[source] = [destination]
   43:         
-  44:         if destination not in self.graph:
+  44:         if destination in self.graph:
   45:             self.graph[destination] = []
   46:     
   47:     def has_cycle(self):
   48:         '''
--------------------------------------------------------------------------------
[0.19995 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_add_edge

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad562c040>

    def test_add_edge(self):
        graph = Graph(5)
        graph.add_edge(1, 2)
>       assert len(graph.graph) == 2
E       AssertionError

ts-3-5/test_3-5_1-0_309.py:38: AssertionError
   - [#  27] COD graph1: 
--------------------------------------------------------------------------------
   63:             elif visited[node] == 1:
   64:                 return True
   65:             visited[node] = -1
   66:             for neighbor in self.graph[node]:
-  67:                 if not (valid(neighbor)):
+  67:                 if valid(neighbor):
   68:                     return False
   69:             return True
   70:         
   71:         for node in range(self.verticies):
--------------------------------------------------------------------------------
[0.20936 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_has_cycle

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad562f3d0>

    def test_has_cycle(self):
        graph = Graph(3)
        graph.add_edge(0, 1)
        graph.add_edge(1, 2)
        assert not graph.has_cycle()
    
        graph.add_edge(2, 0)
>       assert graph.has_cycle()
E       AssertionError

ts-3-5/test_3-5_1-0_309.py:49: AssertionError
   - [#  28] COD graph1: 
--------------------------------------------------------------------------------
   71:         for node in range(self.verticies):
   72:             
   73:             
   74:             
-  75:             if not (visited[node]):
+  75:             if visited[node]:
   76:                 
   77:                 if valid(node):
   78:                     
   79:                     visited = list(map((lambda x: (abs(x))), visited))
--------------------------------------------------------------------------------
[0.21166 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_has_cycle

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad5b4dcc0>

    def test_has_cycle(self):
        graph = Graph(3)
        graph.add_edge(0, 1)
        graph.add_edge(1, 2)
        assert not graph.has_cycle()
    
        graph.add_edge(2, 0)
>       assert graph.has_cycle()
E       AssertionError

ts-3-5/test_3-5_1-0_309.py:49: AssertionError
   - [#  29] COD graph1: 
--------------------------------------------------------------------------------
  109:         
  110:         def dfs(vertex):
  111:             visited.add(vertex)
  112:             for j in self.graph[vertex]:
- 113:                 if j not in visited:
+ 113:                 if j in visited:
  114:                     dfs(j)
  115:             
  116:             stack.appendleft(vertex)
  117:         
--------------------------------------------------------------------------------
[0.21652 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_topological_sort

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad5758130>

    def test_topological_sort(self):
        graph = Graph(4)
        graph.add_edge(0, 1)
        graph.add_edge(1, 2)
        graph.add_edge(1, 3)
        graph.add_edge(2, 3)
>       assert graph.topological_sort() == [0, 1, 2, 3]
E       AssertionError

ts-3-5/test_3-5_1-0_309.py:57: AssertionError
   - [#  30] COD graph1: 
--------------------------------------------------------------------------------
  115:             
  116:             stack.appendleft(vertex)
  117:         
  118:         for (key, _) in self.graph.items():
- 119:             if key not in visited:
+ 119:             if key in visited:
  120:                 dfs(key)
  121:         
  122:         return list(stack)
  123: 
--------------------------------------------------------------------------------
[0.20730 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_topological_sort

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad54e57b0>

    def test_topological_sort(self):
        graph = Graph(4)
        graph.add_edge(0, 1)
        graph.add_edge(1, 2)
        graph.add_edge(1, 3)
        graph.add_edge(2, 3)
>       assert graph.topological_sort() == [0, 1, 2, 3]
E       AssertionError

ts-3-5/test_3-5_1-0_309.py:57: AssertionError
   - [#  31] COD graph1: 
--------------------------------------------------------------------------------
  148:         vertex = WeightedGraphNode(val)
  149:         self.vertices[val] = vertex
  150:     
  151:     def add_edge(self, src, dst, weight=0):
- 152:         if src not in self.vertices:
+ 152:         if src in self.vertices:
  153:             self.add_vertex(src)
  154:         if dst not in self.vertices:
  155:             self.add_vertex(dst)
  156:         self.vertices[src].add_adjacent(dst, weight)
--------------------------------------------------------------------------------
[0.23341 s] killed by ts-3-5/test_3-5_1-0_309.py::TestWeightedGraph::test_add_edge

self = <test_3-5_1-0_309.TestWeightedGraph object at 0x7d3ad5758670>

    def test_add_edge(self):
        w_graph = WeightedGraph()
>       w_graph.add_edge(1, 2, 5)

ts-3-5/test_3-5_1-0_309.py:92: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.WeightedGraph object at 0x7d3ad557bd30>, src = 1, dst = 2
weight = 5

>   ???
E   KeyError: 1

graph1:156: KeyError
   - [#  32] COD graph1: 
--------------------------------------------------------------------------------
  150:     
  151:     def add_edge(self, src, dst, weight=0):
  152:         if src not in self.vertices:
  153:             self.add_vertex(src)
- 154:         if dst not in self.vertices:
+ 154:         if dst in self.vertices:
  155:             self.add_vertex(dst)
  156:         self.vertices[src].add_adjacent(dst, weight)
  157:     
  158:     def remove_edge(self, src, dst):
--------------------------------------------------------------------------------
[0.21045 s] killed by ts-3-5/test_3-5_1-0_309.py::TestWeightedGraph::test_add_edge

self = <test_3-5_1-0_309.TestWeightedGraph object at 0x7d3ad56aa3b0>

    def test_add_edge(self):
        w_graph = WeightedGraph()
        w_graph.add_edge(1, 2, 5)
>       assert len(w_graph.vertices) == 2
E       AssertionError

ts-3-5/test_3-5_1-0_309.py:93: AssertionError
   - [#  33] COD graph1: 
--------------------------------------------------------------------------------
  155:             self.add_vertex(dst)
  156:         self.vertices[src].add_adjacent(dst, weight)
  157:     
  158:     def remove_edge(self, src, dst):
- 159:         if src not in self.vertices:
+ 159:         if src in self.vertices:
  160:             return
  161:         if dst not in self.vertices:
  162:             return
  163:         if dst in self.vertices[src].adjacent:
--------------------------------------------------------------------------------
[0.21350 s] killed by ts-3-5/test_3-5_1-0_309.py::TestWeightedGraph::test_remove_edge

self = <test_3-5_1-0_309.TestWeightedGraph object at 0x7d3ad5b9cd00>

    def test_remove_edge(self):
        w_graph = WeightedGraph()
        w_graph.add_edge(1, 2, 5)
        w_graph.remove_edge(1, 2)
>       assert 2 not in w_graph.vertices[1].adjacent
E       AssertionError

ts-3-5/test_3-5_1-0_309.py:101: AssertionError
   - [#  34] COD graph1: 
--------------------------------------------------------------------------------
  157:     
  158:     def remove_edge(self, src, dst):
  159:         if src not in self.vertices:
  160:             return
- 161:         if dst not in self.vertices:
+ 161:         if dst in self.vertices:
  162:             return
  163:         if dst in self.vertices[src].adjacent:
  164:             self.vertices[src].remove_adjacent(dst)
--------------------------------------------------------------------------------
[0.20266 s] killed by ts-3-5/test_3-5_1-0_309.py::TestWeightedGraph::test_remove_edge

self = <test_3-5_1-0_309.TestWeightedGraph object at 0x7d3ad5b9fbb0>

    def test_remove_edge(self):
        w_graph = WeightedGraph()
        w_graph.add_edge(1, 2, 5)
        w_graph.remove_edge(1, 2)
>       assert 2 not in w_graph.vertices[1].adjacent
E       AssertionError

ts-3-5/test_3-5_1-0_309.py:101: AssertionError
   - [#  35] COI graph1: 
--------------------------------------------------------------------------------
   32:             source: the source vertex
   33:             destination: the destination vertex
   34: 
   35:         '''
-  36:         if len(self.graph) > self.verticies:
+  36:         if not (len(self.graph) > self.verticies):
   37:             raise IndexError('Too many verticies in graph.')
   38:         
   39:         if source in self.graph:
   40:             self.graph[source].append(destination)
--------------------------------------------------------------------------------
[0.20556 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_add_edge

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad5791f30>

    def test_add_edge(self):
        graph = Graph(5)
>       graph.add_edge(1, 2)

ts-3-5/test_3-5_1-0_309.py:37: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.Graph object at 0x7d3ad57936d0>, source = 1, destination = 2

>   ???
E   IndexError: Too many verticies in graph.

graph1:37: IndexError
   - [#  36] COI graph1: 
--------------------------------------------------------------------------------
   35:         '''
   36:         if len(self.graph) > self.verticies:
   37:             raise IndexError('Too many verticies in graph.')
   38:         
-  39:         if source in self.graph:
+  39:         if not (source in self.graph):
   40:             self.graph[source].append(destination)
   41:         else:
   42:             self.graph[source] = [destination]
   43:         
--------------------------------------------------------------------------------
[0.20661 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_add_edge

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad5677d30>

    def test_add_edge(self):
        graph = Graph(5)
>       graph.add_edge(1, 2)

ts-3-5/test_3-5_1-0_309.py:37: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.Graph object at 0x7d3ad5a74670>, source = 1, destination = 2

>   ???
E   KeyError: 1

graph1:40: KeyError
   - [#  37] COI graph1: 
--------------------------------------------------------------------------------
   35:         '''
   36:         if len(self.graph) > self.verticies:
   37:             raise IndexError('Too many verticies in graph.')
   38:         
-  39:         if source in self.graph:
+  39:         if source not in self.graph:
   40:             self.graph[source].append(destination)
   41:         else:
   42:             self.graph[source] = [destination]
   43:         
--------------------------------------------------------------------------------
[0.20755 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_add_edge

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad5610df0>

    def test_add_edge(self):
        graph = Graph(5)
>       graph.add_edge(1, 2)

ts-3-5/test_3-5_1-0_309.py:37: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.Graph object at 0x7d3ad5610490>, source = 1, destination = 2

>   ???
E   KeyError: 1

graph1:40: KeyError
   - [#  38] COI graph1: 
--------------------------------------------------------------------------------
   40:             self.graph[source].append(destination)
   41:         else:
   42:             self.graph[source] = [destination]
   43:         
-  44:         if destination not in self.graph:
+  44:         if not (destination not in self.graph):
   45:             self.graph[destination] = []
   46:     
   47:     def has_cycle(self):
   48:         '''
--------------------------------------------------------------------------------
[0.20697 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_add_edge

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad54dbca0>

    def test_add_edge(self):
        graph = Graph(5)
        graph.add_edge(1, 2)
>       assert len(graph.graph) == 2
E       AssertionError

ts-3-5/test_3-5_1-0_309.py:38: AssertionError
   - [#  39] COI graph1: 
--------------------------------------------------------------------------------
   57:         visited = [0] * self.verticies
   58:         
   59:         def valid(node):
   60:             
-  61:             if visited[node] == -1:
+  61:             if not (visited[node] == -1):
   62:                 return False
   63:             elif visited[node] == 1:
   64:                 return True
   65:             visited[node] = -1
--------------------------------------------------------------------------------
[0.23309 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_has_cycle

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad56dc220>

    def test_has_cycle(self):
        graph = Graph(3)
        graph.add_edge(0, 1)
        graph.add_edge(1, 2)
>       assert not graph.has_cycle()
E       AssertionError

ts-3-5/test_3-5_1-0_309.py:46: AssertionError
   - [#  40] COI graph1: 
--------------------------------------------------------------------------------
   59:         def valid(node):
   60:             
   61:             if visited[node] == -1:
   62:                 return False
-  63:             elif visited[node] == 1:
+  63:             elif not (visited[node] == 1):
   64:                 return True
   65:             visited[node] = -1
   66:             for neighbor in self.graph[node]:
   67:                 if not (valid(neighbor)):
--------------------------------------------------------------------------------
[0.20820 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_has_cycle

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad55d1c60>

    def test_has_cycle(self):
        graph = Graph(3)
        graph.add_edge(0, 1)
        graph.add_edge(1, 2)
        assert not graph.has_cycle()
    
        graph.add_edge(2, 0)
>       assert graph.has_cycle()
E       AssertionError

ts-3-5/test_3-5_1-0_309.py:49: AssertionError
   - [#  41] COI graph1: 
--------------------------------------------------------------------------------
   63:             elif visited[node] == 1:
   64:                 return True
   65:             visited[node] = -1
   66:             for neighbor in self.graph[node]:
-  67:                 if not (valid(neighbor)):
+  67:                 if not ((not (valid(neighbor)))):
   68:                     return False
   69:             return True
   70:         
   71:         for node in range(self.verticies):
--------------------------------------------------------------------------------
[0.22918 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_has_cycle

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad56cd000>

    def test_has_cycle(self):
        graph = Graph(3)
        graph.add_edge(0, 1)
        graph.add_edge(1, 2)
        assert not graph.has_cycle()
    
        graph.add_edge(2, 0)
>       assert graph.has_cycle()
E       AssertionError

ts-3-5/test_3-5_1-0_309.py:49: AssertionError
   - [#  42] COI graph1: 
--------------------------------------------------------------------------------
   71:         for node in range(self.verticies):
   72:             
   73:             
   74:             
-  75:             if not (visited[node]):
+  75:             if not ((not (visited[node]))):
   76:                 
   77:                 if valid(node):
   78:                     
   79:                     visited = list(map((lambda x: (abs(x))), visited))
--------------------------------------------------------------------------------
[0.20834 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_has_cycle

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad5538430>

    def test_has_cycle(self):
        graph = Graph(3)
        graph.add_edge(0, 1)
        graph.add_edge(1, 2)
        assert not graph.has_cycle()
    
        graph.add_edge(2, 0)
>       assert graph.has_cycle()
E       AssertionError

ts-3-5/test_3-5_1-0_309.py:49: AssertionError
   - [#  43] COI graph1: 
--------------------------------------------------------------------------------
   73:             
   74:             
   75:             if not (visited[node]):
   76:                 
-  77:                 if valid(node):
+  77:                 if not (valid(node)):
   78:                     
   79:                     visited = list(map((lambda x: (abs(x))), visited))
   80:                 else:
   81:                     
--------------------------------------------------------------------------------
[0.21305 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_has_cycle

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad5b4da20>

    def test_has_cycle(self):
        graph = Graph(3)
        graph.add_edge(0, 1)
        graph.add_edge(1, 2)
>       assert not graph.has_cycle()
E       AssertionError

ts-3-5/test_3-5_1-0_309.py:46: AssertionError
   - [#  44] COI graph1: 
--------------------------------------------------------------------------------
  109:         
  110:         def dfs(vertex):
  111:             visited.add(vertex)
  112:             for j in self.graph[vertex]:
- 113:                 if j not in visited:
+ 113:                 if not (j not in visited):
  114:                     dfs(j)
  115:             
  116:             stack.appendleft(vertex)
  117:         
--------------------------------------------------------------------------------
[0.20723 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_topological_sort

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad5695690>

    def test_topological_sort(self):
        graph = Graph(4)
        graph.add_edge(0, 1)
        graph.add_edge(1, 2)
        graph.add_edge(1, 3)
        graph.add_edge(2, 3)
>       assert graph.topological_sort() == [0, 1, 2, 3]
E       AssertionError

ts-3-5/test_3-5_1-0_309.py:57: AssertionError
   - [#  45] COI graph1: 
--------------------------------------------------------------------------------
  115:             
  116:             stack.appendleft(vertex)
  117:         
  118:         for (key, _) in self.graph.items():
- 119:             if key not in visited:
+ 119:             if not (key not in visited):
  120:                 dfs(key)
  121:         
  122:         return list(stack)
  123: 
--------------------------------------------------------------------------------
[0.20767 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_topological_sort

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad56cc520>

    def test_topological_sort(self):
        graph = Graph(4)
        graph.add_edge(0, 1)
        graph.add_edge(1, 2)
        graph.add_edge(1, 3)
        graph.add_edge(2, 3)
>       assert graph.topological_sort() == [0, 1, 2, 3]
E       AssertionError

ts-3-5/test_3-5_1-0_309.py:57: AssertionError
   - [#  46] COI graph1: 
--------------------------------------------------------------------------------
  148:         vertex = WeightedGraphNode(val)
  149:         self.vertices[val] = vertex
  150:     
  151:     def add_edge(self, src, dst, weight=0):
- 152:         if src not in self.vertices:
+ 152:         if not (src not in self.vertices):
  153:             self.add_vertex(src)
  154:         if dst not in self.vertices:
  155:             self.add_vertex(dst)
  156:         self.vertices[src].add_adjacent(dst, weight)
--------------------------------------------------------------------------------
[0.20829 s] killed by ts-3-5/test_3-5_1-0_309.py::TestWeightedGraph::test_add_edge

self = <test_3-5_1-0_309.TestWeightedGraph object at 0x7d3ad556a4d0>

    def test_add_edge(self):
        w_graph = WeightedGraph()
>       w_graph.add_edge(1, 2, 5)

ts-3-5/test_3-5_1-0_309.py:92: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.WeightedGraph object at 0x7d3ad556a8c0>, src = 1, dst = 2
weight = 5

>   ???
E   KeyError: 1

graph1:156: KeyError
   - [#  47] COI graph1: 
--------------------------------------------------------------------------------
  150:     
  151:     def add_edge(self, src, dst, weight=0):
  152:         if src not in self.vertices:
  153:             self.add_vertex(src)
- 154:         if dst not in self.vertices:
+ 154:         if not (dst not in self.vertices):
  155:             self.add_vertex(dst)
  156:         self.vertices[src].add_adjacent(dst, weight)
  157:     
  158:     def remove_edge(self, src, dst):
--------------------------------------------------------------------------------
[0.21496 s] killed by ts-3-5/test_3-5_1-0_309.py::TestWeightedGraph::test_add_edge

self = <test_3-5_1-0_309.TestWeightedGraph object at 0x7d3ad565f7f0>

    def test_add_edge(self):
        w_graph = WeightedGraph()
        w_graph.add_edge(1, 2, 5)
>       assert len(w_graph.vertices) == 2
E       AssertionError

ts-3-5/test_3-5_1-0_309.py:93: AssertionError
   - [#  48] COI graph1: 
--------------------------------------------------------------------------------
  155:             self.add_vertex(dst)
  156:         self.vertices[src].add_adjacent(dst, weight)
  157:     
  158:     def remove_edge(self, src, dst):
- 159:         if src not in self.vertices:
+ 159:         if not (src not in self.vertices):
  160:             return
  161:         if dst not in self.vertices:
  162:             return
  163:         if dst in self.vertices[src].adjacent:
--------------------------------------------------------------------------------
[0.21309 s] killed by ts-3-5/test_3-5_1-0_309.py::TestWeightedGraph::test_remove_edge

self = <test_3-5_1-0_309.TestWeightedGraph object at 0x7d3ad5bf2290>

    def test_remove_edge(self):
        w_graph = WeightedGraph()
        w_graph.add_edge(1, 2, 5)
        w_graph.remove_edge(1, 2)
>       assert 2 not in w_graph.vertices[1].adjacent
E       AssertionError

ts-3-5/test_3-5_1-0_309.py:101: AssertionError
   - [#  49] COI graph1: 
--------------------------------------------------------------------------------
  157:     
  158:     def remove_edge(self, src, dst):
  159:         if src not in self.vertices:
  160:             return
- 161:         if dst not in self.vertices:
+ 161:         if not (dst not in self.vertices):
  162:             return
  163:         if dst in self.vertices[src].adjacent:
  164:             self.vertices[src].remove_adjacent(dst)
--------------------------------------------------------------------------------
[0.20927 s] killed by ts-3-5/test_3-5_1-0_309.py::TestWeightedGraph::test_remove_edge

self = <test_3-5_1-0_309.TestWeightedGraph object at 0x7d3ad575a2c0>

    def test_remove_edge(self):
        w_graph = WeightedGraph()
        w_graph.add_edge(1, 2, 5)
        w_graph.remove_edge(1, 2)
>       assert 2 not in w_graph.vertices[1].adjacent
E       AssertionError

ts-3-5/test_3-5_1-0_309.py:101: AssertionError
   - [#  50] COI graph1: 
--------------------------------------------------------------------------------
  159:         if src not in self.vertices:
  160:             return
  161:         if dst not in self.vertices:
  162:             return
- 163:         if dst in self.vertices[src].adjacent:
+ 163:         if not (dst in self.vertices[src].adjacent):
  164:             self.vertices[src].remove_adjacent(dst)
--------------------------------------------------------------------------------
[0.21137 s] killed by ts-3-5/test_3-5_1-0_309.py::TestWeightedGraph::test_remove_edge

self = <test_3-5_1-0_309.TestWeightedGraph object at 0x7d3ad57920b0>

    def test_remove_edge(self):
        w_graph = WeightedGraph()
        w_graph.add_edge(1, 2, 5)
        w_graph.remove_edge(1, 2)
>       assert 2 not in w_graph.vertices[1].adjacent
E       AssertionError

ts-3-5/test_3-5_1-0_309.py:101: AssertionError
   - [#  51] COI graph1: 
--------------------------------------------------------------------------------
  159:         if src not in self.vertices:
  160:             return
  161:         if dst not in self.vertices:
  162:             return
- 163:         if dst in self.vertices[src].adjacent:
+ 163:         if dst not in self.vertices[src].adjacent:
  164:             self.vertices[src].remove_adjacent(dst)
--------------------------------------------------------------------------------
[0.20558 s] killed by ts-3-5/test_3-5_1-0_309.py::TestWeightedGraph::test_remove_edge

self = <test_3-5_1-0_309.TestWeightedGraph object at 0x7d3ad57fdbd0>

    def test_remove_edge(self):
        w_graph = WeightedGraph()
        w_graph.add_edge(1, 2, 5)
        w_graph.remove_edge(1, 2)
>       assert 2 not in w_graph.vertices[1].adjacent
E       AssertionError

ts-3-5/test_3-5_1-0_309.py:101: AssertionError
   - [#  52] OIL graph1: 
--------------------------------------------------------------------------------
   65:             visited[node] = -1
   66:             for neighbor in self.graph[node]:
   67:                 if not (valid(neighbor)):
   68:                     return False
-  69:             return True
-  70:         
+  69:                 break
+  70:             return True
   71:         for node in range(self.verticies):
   72:             
   73:             
   74:             
--------------------------------------------------------------------------------
[0.18507 s] survived
   - [#  53] OIL graph1: 
--------------------------------------------------------------------------------
   78:                     
   79:                     visited = list(map((lambda x: (abs(x))), visited))
   80:                 else:
   81:                     
-  82:                     return True
+  82:                     return Truebreak
   83:         
   84:         return False
   85:     
   86:     def topological_sort(self):
--------------------------------------------------------------------------------
[0.19518 s] survived
   - [#  54] OIL graph1: 
--------------------------------------------------------------------------------
  111:             visited.add(vertex)
  112:             for j in self.graph[vertex]:
  113:                 if j not in visited:
  114:                     dfs(j)
- 115:             
+ 115:                 break
  116:             stack.appendleft(vertex)
  117:         
  118:         for (key, _) in self.graph.items():
  119:             if key not in visited:
--------------------------------------------------------------------------------
[0.16460 s] survived
   - [#  55] OIL graph1: 
--------------------------------------------------------------------------------
  117:         
  118:         for (key, _) in self.graph.items():
  119:             if key not in visited:
  120:                 dfs(key)
- 121:         
+ 121:             break
  122:         return list(stack)
  123: 
  124: class WeightedGraphNode:
  125:     
--------------------------------------------------------------------------------
[0.18453 s] survived
   - [#  56] RIL graph1: 
--------------------------------------------------------------------------------
   62:                 return False
   63:             elif visited[node] == 1:
   64:                 return True
   65:             visited[node] = -1
-  66:             for neighbor in self.graph[node]:
+  66:             for neighbor in reversed(self.graph[node]):
   67:                 if not (valid(neighbor)):
   68:                     return False
   69:             return True
   70:         
--------------------------------------------------------------------------------
[0.18418 s] survived
   - [#  57] RIL graph1: 
--------------------------------------------------------------------------------
   67:                 if not (valid(neighbor)):
   68:                     return False
   69:             return True
   70:         
-  71:         for node in range(self.verticies):
+  71:         for node in reversed(range(self.verticies)):
   72:             
   73:             
   74:             
   75:             if not (visited[node]):
--------------------------------------------------------------------------------
[0.18696 s] survived
   - [#  58] RIL graph1: 
--------------------------------------------------------------------------------
  108:         stack = deque()
  109:         
  110:         def dfs(vertex):
  111:             visited.add(vertex)
- 112:             for j in self.graph[vertex]:
+ 112:             for j in reversed(self.graph[vertex]):
  113:                 if j not in visited:
  114:                     dfs(j)
  115:             
  116:             stack.appendleft(vertex)
--------------------------------------------------------------------------------
[0.18482 s] survived
   - [#  59] RIL graph1: 
--------------------------------------------------------------------------------
  114:                     dfs(j)
  115:             
  116:             stack.appendleft(vertex)
  117:         
- 118:         for (key, _) in self.graph.items():
+ 118:         for (key, _) in reversed(self.graph.items()):
  119:             if key not in visited:
  120:                 dfs(key)
  121:         
  122:         return list(stack)
--------------------------------------------------------------------------------
[0.19409 s] survived
   - [#  60] ROR graph1: 
--------------------------------------------------------------------------------
   32:             source: the source vertex
   33:             destination: the destination vertex
   34: 
   35:         '''
-  36:         if len(self.graph) > self.verticies:
+  36:         if len(self.graph) < self.verticies:
   37:             raise IndexError('Too many verticies in graph.')
   38:         
   39:         if source in self.graph:
   40:             self.graph[source].append(destination)
--------------------------------------------------------------------------------
[0.20033 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_add_edge

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad56b9900>

    def test_add_edge(self):
        graph = Graph(5)
>       graph.add_edge(1, 2)

ts-3-5/test_3-5_1-0_309.py:37: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.Graph object at 0x7d3ad56ba200>, source = 1, destination = 2

>   ???
E   IndexError: Too many verticies in graph.

graph1:37: IndexError
   - [#  61] ROR graph1: 
--------------------------------------------------------------------------------
   32:             source: the source vertex
   33:             destination: the destination vertex
   34: 
   35:         '''
-  36:         if len(self.graph) > self.verticies:
+  36:         if len(self.graph) >= self.verticies:
   37:             raise IndexError('Too many verticies in graph.')
   38:         
   39:         if source in self.graph:
   40:             self.graph[source].append(destination)
--------------------------------------------------------------------------------
[0.22858 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_has_cycle

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad55aab00>

    def test_has_cycle(self):
        graph = Graph(3)
        graph.add_edge(0, 1)
        graph.add_edge(1, 2)
        assert not graph.has_cycle()
    
>       graph.add_edge(2, 0)

ts-3-5/test_3-5_1-0_309.py:48: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.Graph object at 0x7d3ad57916c0>, source = 2, destination = 0

>   ???
E   IndexError: Too many verticies in graph.

graph1:37: IndexError
   - [#  62] ROR graph1: 
--------------------------------------------------------------------------------
   57:         visited = [0] * self.verticies
   58:         
   59:         def valid(node):
   60:             
-  61:             if visited[node] == -1:
+  61:             if visited[node] != -1:
   62:                 return False
   63:             elif visited[node] == 1:
   64:                 return True
   65:             visited[node] = -1
--------------------------------------------------------------------------------
[0.20924 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_has_cycle

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad5b9f9d0>

    def test_has_cycle(self):
        graph = Graph(3)
        graph.add_edge(0, 1)
        graph.add_edge(1, 2)
>       assert not graph.has_cycle()
E       AssertionError

ts-3-5/test_3-5_1-0_309.py:46: AssertionError
   - [#  63] ROR graph1: 
--------------------------------------------------------------------------------
   59:         def valid(node):
   60:             
   61:             if visited[node] == -1:
   62:                 return False
-  63:             elif visited[node] == 1:
+  63:             elif visited[node] != 1:
   64:                 return True
   65:             visited[node] = -1
   66:             for neighbor in self.graph[node]:
   67:                 if not (valid(neighbor)):
--------------------------------------------------------------------------------
[0.20784 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_has_cycle

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad56cfa90>

    def test_has_cycle(self):
        graph = Graph(3)
        graph.add_edge(0, 1)
        graph.add_edge(1, 2)
        assert not graph.has_cycle()
    
        graph.add_edge(2, 0)
>       assert graph.has_cycle()
E       AssertionError

ts-3-5/test_3-5_1-0_309.py:49: AssertionError
   - [#  64] SDI graph1: 
--------------------------------------------------------------------------------
    1: from collections import defaultdict, deque
    2: 
    3: class GraphNode:
    4:     
-   5:     def __init__(self, val):
-   6:         self.val = val
-   7:         self.adjacent_list = set()
-   8:     
+   5:     @staticmethod
+   6:     def __init__(self, val):
+   7:         self.val = val
+   8:         self.adjacent_list = set()
    9:     def add_adjacent(self, node):
   10:         self.adjacent_list.add(node)
   11:     
   12:     def remove_adjacent(self, node):
--------------------------------------------------------------------------------
[0.20600 s] incompetent

TypeError: /home/lucca/teste/python_experiments2/graph1/ts-3-5/test_3-5_1-0_309.py:16: TypeError: GraphNode.__init__() missing 1 required positional argument: 'val'

   - [#  65] SDI graph1: 
--------------------------------------------------------------------------------
    5:     def __init__(self, val):
    6:         self.val = val
    7:         self.adjacent_list = set()
    8:     
-   9:     def add_adjacent(self, node):
-  10:         self.adjacent_list.add(node)
-  11:     
+   9:     @staticmethod
+  10:     def add_adjacent(self, node):
+  11:         self.adjacent_list.add(node)
   12:     def remove_adjacent(self, node):
   13:         self.adjacent_list.remove(node)
   14: 
   15: class Graph:
--------------------------------------------------------------------------------
[0.20886 s] incompetent

TypeError: /home/lucca/teste/python_experiments2/graph1/ts-3-5/test_3-5_1-0_309.py:23: TypeError: GraphNode.add_adjacent() missing 1 required positional argument: 'node'

   - [#  66] SDI graph1: 
--------------------------------------------------------------------------------
    8:     
    9:     def add_adjacent(self, node):
   10:         self.adjacent_list.add(node)
   11:     
-  12:     def remove_adjacent(self, node):
-  13:         self.adjacent_list.remove(node)
-  14: 
+  12:     @staticmethod
+  13:     def remove_adjacent(self, node):
+  14:         self.adjacent_list.remove(node)
   15: class Graph:
   16:     '''
   17:     A directed graph represented with an adjacency list.
   18:     '''
--------------------------------------------------------------------------------
[0.18502 s] survived
   - [#  67] SDI graph1: 
--------------------------------------------------------------------------------
   16:     '''
   17:     A directed graph represented with an adjacency list.
   18:     '''
   19:     
-  20:     def __init__(self, verticies):
-  21:         self.graph = {}
-  22:         self.verticies = verticies
-  23:     
+  20:     @staticmethod
+  21:     def __init__(self, verticies):
+  22:         self.graph = {}
+  23:         self.verticies = verticies
   24:     def add_edge(self, source, destination):
   25:         '''
   26:         Add an edge to the graph.
   27: 
--------------------------------------------------------------------------------
[0.20449 s] incompetent

TypeError: /home/lucca/teste/python_experiments2/graph1/ts-3-5/test_3-5_1-0_309.py:31: TypeError: Graph.__init__() missing 1 required positional argument: 'verticies'

   - [#  68] SDI graph1: 
--------------------------------------------------------------------------------
   20:     def __init__(self, verticies):
   21:         self.graph = {}
   22:         self.verticies = verticies
   23:     
-  24:     def add_edge(self, source, destination):
-  25:         '''
-  26:         Add an edge to the graph.
-  27: 
-  28:         Add an edge pointing from source vertex
-  29:         to destination vertex.
-  30: 
-  31:         Args:
-  32:             source: the source vertex
-  33:             destination: the destination vertex
-  34: 
-  35:         '''
-  36:         if len(self.graph) > self.verticies:
-  37:             raise IndexError('Too many verticies in graph.')
-  38:         
-  39:         if source in self.graph:
-  40:             self.graph[source].append(destination)
-  41:         else:
-  42:             self.graph[source] = [destination]
-  43:         
-  44:         if destination not in self.graph:
-  45:             self.graph[destination] = []
-  46:     
+  24:     @staticmethod
+  25:     def add_edge(self, source, destination):
+  26:         '''
+  27:         Add an edge to the graph.
+  28: 
+  29:         Add an edge pointing from source vertex
+  30:         to destination vertex.
+  31: 
+  32:         Args:
+  33:             source: the source vertex
+  34:             destination: the destination vertex
+  35: 
+  36:         '''
+  37:         if len(self.graph) > self.verticies:
+  38:             raise IndexError('Too many verticies in graph.')
+  39:         
+  40:         if source in self.graph:
+  41:             self.graph[source].append(destination)
+  42:         else:
+  43:             self.graph[source] = [destination]
+  44:         
+  45:         if destination not in self.graph:
+  46:             self.graph[destination] = []
   47:     def has_cycle(self):
   48:         '''
   49:         Detect if a graph has a cycle.
   50: 
--------------------------------------------------------------------------------
[0.20415 s] incompetent

TypeError: /home/lucca/teste/python_experiments2/graph1/ts-3-5/test_3-5_1-0_309.py:37: TypeError: Graph.add_edge() missing 1 required positional argument: 'destination'

   - [#  69] SDI graph1: 
--------------------------------------------------------------------------------
   43:         
   44:         if destination not in self.graph:
   45:             self.graph[destination] = []
   46:     
-  47:     def has_cycle(self):
-  48:         '''
-  49:         Detect if a graph has a cycle.
-  50: 
-  51:         Returns:
-  52:             True if the graph has a cycle and
-  53:             False if the graph is acyclic.
-  54: 
-  55:         '''
-  56:         
-  57:         visited = [0] * self.verticies
-  58:         
-  59:         def valid(node):
-  60:             
-  61:             if visited[node] == -1:
-  62:                 return False
-  63:             elif visited[node] == 1:
-  64:                 return True
-  65:             visited[node] = -1
-  66:             for neighbor in self.graph[node]:
-  67:                 if not (valid(neighbor)):
-  68:                     return False
-  69:             return True
-  70:         
-  71:         for node in range(self.verticies):
-  72:             
+  47:     @staticmethod
+  48:     def has_cycle(self):
+  49:         '''
+  50:         Detect if a graph has a cycle.
+  51: 
+  52:         Returns:
+  53:             True if the graph has a cycle and
+  54:             False if the graph is acyclic.
+  55: 
+  56:         '''
+  57:         
+  58:         visited = [0] * self.verticies
+  59:         
+  60:         def valid(node):
+  61:             
+  62:             if visited[node] == -1:
+  63:                 return False
+  64:             elif visited[node] == 1:
+  65:                 return True
+  66:             visited[node] = -1
+  67:             for neighbor in self.graph[node]:
+  68:                 if not (valid(neighbor)):
+  69:                     return False
+  70:             return True
+  71:         
+  72:         for node in range(self.verticies):
   73:             
   74:             
-  75:             if not (visited[node]):
-  76:                 
-  77:                 if valid(node):
-  78:                     
-  79:                     visited = list(map((lambda x: (abs(x))), visited))
-  80:                 else:
-  81:                     
-  82:                     return True
-  83:         
-  84:         return False
-  85:     
+  75:             
+  76:             if not (visited[node]):
+  77:                 
+  78:                 if valid(node):
+  79:                     
+  80:                     visited = list(map((lambda x: (abs(x))), visited))
+  81:                 else:
+  82:                     
+  83:                     return True
+  84:         
+  85:         return False
   86:     def topological_sort(self):
   87:         """
   88:         Sort the graph topologically.
   89: 
--------------------------------------------------------------------------------
[0.21183 s] incompetent

TypeError: /home/lucca/teste/python_experiments2/graph1/ts-3-5/test_3-5_1-0_309.py:46: TypeError: Graph.has_cycle() missing 1 required positional argument: 'self'

   - [#  70] SDI graph1: 
--------------------------------------------------------------------------------
   82:                     return True
   83:         
   84:         return False
   85:     
-  86:     def topological_sort(self):
-  87:         """
-  88:         Sort the graph topologically.
-  89: 
-  90:         A topological sort lists nodes in such a way
-  91:         that every node 's' in 's' -> 'd' directed pairs
-  92:         is listed before 'd.'  This will not work in a 
-  93:         graph that contains cycles.
-  94: 
-  95:         The algorithm looks at every node, and does a
-  96:         dfs for each node adjacent to the node and then adds
-  97:         the originating node to a stack once all adjacent
-  98:         nodes have been searched.  In the end, the stack
-  99:         will be in order of a possible topological sort.
- 100: 
- 101:         Topological sorts are not necessarily unique.
- 102: 
- 103:         Returns:
- 104:             A list of vertices in a topological ordering.
- 105: 
- 106:         """
- 107:         visited = set()
- 108:         stack = deque()
- 109:         
- 110:         def dfs(vertex):
- 111:             visited.add(vertex)
- 112:             for j in self.graph[vertex]:
- 113:                 if j not in visited:
- 114:                     dfs(j)
- 115:             
- 116:             stack.appendleft(vertex)
- 117:         
- 118:         for (key, _) in self.graph.items():
- 119:             if key not in visited:
- 120:                 dfs(key)
- 121:         
- 122:         return list(stack)
- 123: 
+  86:     @staticmethod
+  87:     def topological_sort(self):
+  88:         """
+  89:         Sort the graph topologically.
+  90: 
+  91:         A topological sort lists nodes in such a way
+  92:         that every node 's' in 's' -> 'd' directed pairs
+  93:         is listed before 'd.'  This will not work in a 
+  94:         graph that contains cycles.
+  95: 
+  96:         The algorithm looks at every node, and does a
+  97:         dfs for each node adjacent to the node and then adds
+  98:         the originating node to a stack once all adjacent
+  99:         nodes have been searched.  In the end, the stack
+ 100:         will be in order of a possible topological sort.
+ 101: 
+ 102:         Topological sorts are not necessarily unique.
+ 103: 
+ 104:         Returns:
+ 105:             A list of vertices in a topological ordering.
+ 106: 
+ 107:         """
+ 108:         visited = set()
+ 109:         stack = deque()
+ 110:         
+ 111:         def dfs(vertex):
+ 112:             visited.add(vertex)
+ 113:             for j in self.graph[vertex]:
+ 114:                 if j not in visited:
+ 115:                     dfs(j)
+ 116:             
+ 117:             stack.appendleft(vertex)
+ 118:         
+ 119:         for (key, _) in self.graph.items():
+ 120:             if key not in visited:
+ 121:                 dfs(key)
+ 122:         
+ 123:         return list(stack)
  124: class WeightedGraphNode:
  125:     
  126:     def __init__(self, val):
  127:         self.val = val
--------------------------------------------------------------------------------
[0.20960 s] incompetent

TypeError: /home/lucca/teste/python_experiments2/graph1/ts-3-5/test_3-5_1-0_309.py:57: TypeError: Graph.topological_sort() missing 1 required positional argument: 'self'

   - [#  71] SDI graph1: 
--------------------------------------------------------------------------------
  122:         return list(stack)
  123: 
  124: class WeightedGraphNode:
  125:     
- 126:     def __init__(self, val):
- 127:         self.val = val
- 128:         self.adjacent = {}
- 129:     
+ 126:     @staticmethod
+ 127:     def __init__(self, val):
+ 128:         self.val = val
+ 129:         self.adjacent = {}
  130:     def __str__(self):
  131:         return (str(self.val) + ' adjacent: ') + str([x for x in self.adjacent])
  132:     
  133:     def add_adjacent(self, node, weight=0):
--------------------------------------------------------------------------------
[0.20838 s] incompetent

TypeError: /home/lucca/teste/python_experiments2/graph1/ts-3-5/test_3-5_1-0_309.py:64: TypeError: WeightedGraphNode.__init__() missing 1 required positional argument: 'val'

   - [#  72] SDI graph1: 
--------------------------------------------------------------------------------
  126:     def __init__(self, val):
  127:         self.val = val
  128:         self.adjacent = {}
  129:     
- 130:     def __str__(self):
- 131:         return (str(self.val) + ' adjacent: ') + str([x for x in self.adjacent])
- 132:     
+ 130:     @staticmethod
+ 131:     def __str__(self):
+ 132:         return (str(self.val) + ' adjacent: ') + str([x for x in self.adjacent])
  133:     def add_adjacent(self, node, weight=0):
  134:         self.adjacent[node] = weight
  135:     
  136:     def remove_adjacent(self, node):
--------------------------------------------------------------------------------
[0.18462 s] survived
   - [#  73] SDI graph1: 
--------------------------------------------------------------------------------
  129:     
  130:     def __str__(self):
  131:         return (str(self.val) + ' adjacent: ') + str([x for x in self.adjacent])
  132:     
- 133:     def add_adjacent(self, node, weight=0):
- 134:         self.adjacent[node] = weight
- 135:     
+ 133:     @staticmethod
+ 134:     def add_adjacent(self, node, weight=0):
+ 135:         self.adjacent[node] = weight
  136:     def remove_adjacent(self, node):
  137:         del self.adjacent[node]
  138: 
  139: class WeightedGraph:
--------------------------------------------------------------------------------
[0.20418 s] killed by ts-3-5/test_3-5_1-0_309.py::TestWeightedGraphNode::test_add_adjacent

self = <test_3-5_1-0_309.TestWeightedGraphNode object at 0x7d3ad57be0b0>

    def test_add_adjacent(self):
        w_node1 = WeightedGraphNode(10)
        w_node2 = WeightedGraphNode(20)
        w_node1.add_adjacent(w_node2, 5)
>       assert len(w_node1.adjacent) == 1
E       AssertionError

ts-3-5/test_3-5_1-0_309.py:72: AssertionError
   - [#  74] SDI graph1: 
--------------------------------------------------------------------------------
  132:     
  133:     def add_adjacent(self, node, weight=0):
  134:         self.adjacent[node] = weight
  135:     
- 136:     def remove_adjacent(self, node):
- 137:         del self.adjacent[node]
- 138: 
+ 136:     @staticmethod
+ 137:     def remove_adjacent(self, node):
+ 138:         del self.adjacent[node]
  139: class WeightedGraph:
  140:     
  141:     def __init__(self):
  142:         self.vertices = {}
--------------------------------------------------------------------------------
[0.20826 s] incompetent

TypeError: /home/lucca/teste/python_experiments2/graph1/ts-3-5/test_3-5_1-0_309.py:79: TypeError: WeightedGraphNode.remove_adjacent() missing 1 required positional argument: 'node'

   - [#  75] SDI graph1: 
--------------------------------------------------------------------------------
  137:         del self.adjacent[node]
  138: 
  139: class WeightedGraph:
  140:     
- 141:     def __init__(self):
- 142:         self.vertices = {}
- 143:     
+ 141:     @staticmethod
+ 142:     def __init__(self):
+ 143:         self.vertices = {}
  144:     def __str__(self):
  145:         return ', '.join([node.__str__() for node in self.vertices.values()])
  146:     
  147:     def add_vertex(self, val):
--------------------------------------------------------------------------------
[0.21038 s] incompetent

TypeError: /home/lucca/teste/python_experiments2/graph1/ts-3-5/test_3-5_1-0_309.py:87: TypeError: WeightedGraph.__init__() missing 1 required positional argument: 'self'

   - [#  76] SDI graph1: 
--------------------------------------------------------------------------------
  140:     
  141:     def __init__(self):
  142:         self.vertices = {}
  143:     
- 144:     def __str__(self):
- 145:         return ', '.join([node.__str__() for node in self.vertices.values()])
- 146:     
+ 144:     @staticmethod
+ 145:     def __str__(self):
+ 146:         return ', '.join([node.__str__() for node in self.vertices.values()])
  147:     def add_vertex(self, val):
  148:         vertex = WeightedGraphNode(val)
  149:         self.vertices[val] = vertex
  150:     
--------------------------------------------------------------------------------
[0.18402 s] survived
   - [#  77] SDI graph1: 
--------------------------------------------------------------------------------
  143:     
  144:     def __str__(self):
  145:         return ', '.join([node.__str__() for node in self.vertices.values()])
  146:     
- 147:     def add_vertex(self, val):
- 148:         vertex = WeightedGraphNode(val)
- 149:         self.vertices[val] = vertex
- 150:     
+ 147:     @staticmethod
+ 148:     def add_vertex(self, val):
+ 149:         vertex = WeightedGraphNode(val)
+ 150:         self.vertices[val] = vertex
  151:     def add_edge(self, src, dst, weight=0):
  152:         if src not in self.vertices:
  153:             self.add_vertex(src)
  154:         if dst not in self.vertices:
--------------------------------------------------------------------------------
[0.22606 s] incompetent

TypeError: graph1:153: TypeError: WeightedGraph.add_vertex() missing 1 required positional argument: 'val'

   - [#  78] SDI graph1: 
--------------------------------------------------------------------------------
  147:     def add_vertex(self, val):
  148:         vertex = WeightedGraphNode(val)
  149:         self.vertices[val] = vertex
  150:     
- 151:     def add_edge(self, src, dst, weight=0):
- 152:         if src not in self.vertices:
- 153:             self.add_vertex(src)
- 154:         if dst not in self.vertices:
- 155:             self.add_vertex(dst)
- 156:         self.vertices[src].add_adjacent(dst, weight)
- 157:     
+ 151:     @staticmethod
+ 152:     def add_edge(self, src, dst, weight=0):
+ 153:         if src not in self.vertices:
+ 154:             self.add_vertex(src)
+ 155:         if dst not in self.vertices:
+ 156:             self.add_vertex(dst)
+ 157:         self.vertices[src].add_adjacent(dst, weight)
  158:     def remove_edge(self, src, dst):
  159:         if src not in self.vertices:
  160:             return
  161:         if dst not in self.vertices:
--------------------------------------------------------------------------------
[0.20841 s] killed by ts-3-5/test_3-5_1-0_309.py::TestWeightedGraph::test_add_edge

self = <test_3-5_1-0_309.TestWeightedGraph object at 0x7d3ad5851d20>

    def test_add_edge(self):
        w_graph = WeightedGraph()
>       w_graph.add_edge(1, 2, 5)

ts-3-5/test_3-5_1-0_309.py:92: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = 1, src = 2, dst = 5, weight = 0

>   ???
E   AttributeError: 'int' object has no attribute 'vertices'

graph1:153: AttributeError
   - [#  79] SDI graph1: 
--------------------------------------------------------------------------------
  154:         if dst not in self.vertices:
  155:             self.add_vertex(dst)
  156:         self.vertices[src].add_adjacent(dst, weight)
  157:     
- 158:     def remove_edge(self, src, dst):
- 159:         if src not in self.vertices:
- 160:             return
- 161:         if dst not in self.vertices:
- 162:             return
- 163:         if dst in self.vertices[src].adjacent:
- 164:             self.vertices[src].remove_adjacent(dst)
+ 158:     @staticmethod
+ 159:     def remove_edge(self, src, dst):
+ 160:         if src not in self.vertices:
+ 161:             return
+ 162:         if dst not in self.vertices:
+ 163:             return
+ 164:         if dst in self.vertices[src].adjacent:
+ 165:             self.vertices[src].remove_adjacent(dst)
--------------------------------------------------------------------------------
[0.21373 s] incompetent

TypeError: /home/lucca/teste/python_experiments2/graph1/ts-3-5/test_3-5_1-0_309.py:100: TypeError: WeightedGraph.remove_edge() missing 1 required positional argument: 'dst'

   - [#  80] SDL graph1: 
--------------------------------------------------------------------------------
    2: 
    3: class GraphNode:
    4:     
    5:     def __init__(self, val):
-   6:         self.val = val
+   6:         pass
    7:         self.adjacent_list = set()
    8:     
    9:     def add_adjacent(self, node):
   10:         self.adjacent_list.add(node)
--------------------------------------------------------------------------------
[0.20344 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraphNode::test_init

self = <test_3-5_1-0_309.TestGraphNode object at 0x7d3ad572d840>

    def test_init(self):
        node = GraphNode(1)
>       assert node.val == 1
E       AttributeError: 'GraphNode' object has no attribute 'val'

ts-3-5/test_3-5_1-0_309.py:17: AttributeError
   - [#  81] SDL graph1: 
--------------------------------------------------------------------------------
    3: class GraphNode:
    4:     
    5:     def __init__(self, val):
    6:         self.val = val
-   7:         self.adjacent_list = set()
+   7:         pass
    8:     
    9:     def add_adjacent(self, node):
   10:         self.adjacent_list.add(node)
   11:     
--------------------------------------------------------------------------------
[0.20937 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraphNode::test_init

self = <test_3-5_1-0_309.TestGraphNode object at 0x7d3ad5bf32b0>

    def test_init(self):
        node = GraphNode(1)
        assert node.val == 1
>       assert len(node.adjacent_list) == 0
E       AttributeError: 'GraphNode' object has no attribute 'adjacent_list'

ts-3-5/test_3-5_1-0_309.py:18: AttributeError
   - [#  82] SDL graph1: 
--------------------------------------------------------------------------------
    6:         self.val = val
    7:         self.adjacent_list = set()
    8:     
    9:     def add_adjacent(self, node):
-  10:         self.adjacent_list.add(node)
+  10:         pass
   11:     
   12:     def remove_adjacent(self, node):
   13:         self.adjacent_list.remove(node)
   14: 
--------------------------------------------------------------------------------
[0.20503 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraphNode::test_add_adjacent

self = <test_3-5_1-0_309.TestGraphNode object at 0x7d3ad57e57e0>

    def test_add_adjacent(self):
        node1 = GraphNode(1)
        node2 = GraphNode(2)
        node1.add_adjacent(node2)
>       assert len(node1.adjacent_list) == 1
E       AssertionError

ts-3-5/test_3-5_1-0_309.py:24: AssertionError
   - [#  83] SDL graph1: 
--------------------------------------------------------------------------------
    9:     def add_adjacent(self, node):
   10:         self.adjacent_list.add(node)
   11:     
   12:     def remove_adjacent(self, node):
-  13:         self.adjacent_list.remove(node)
+  13:         pass
   14: 
   15: class Graph:
   16:     '''
   17:     A directed graph represented with an adjacency list.
--------------------------------------------------------------------------------
[0.18437 s] survived
   - [#  84] SDL graph1: 
--------------------------------------------------------------------------------
   17:     A directed graph represented with an adjacency list.
   18:     '''
   19:     
   20:     def __init__(self, verticies):
-  21:         self.graph = {}
+  21:         pass
   22:         self.verticies = verticies
   23:     
   24:     def add_edge(self, source, destination):
   25:         '''
--------------------------------------------------------------------------------
[0.20712 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_init

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad55ffe50>

    def test_init(self):
        graph = Graph(5)
>       assert len(graph.graph) == 0
E       AttributeError: 'Graph' object has no attribute 'graph'

ts-3-5/test_3-5_1-0_309.py:32: AttributeError
   - [#  85] SDL graph1: 
--------------------------------------------------------------------------------
   18:     '''
   19:     
   20:     def __init__(self, verticies):
   21:         self.graph = {}
-  22:         self.verticies = verticies
+  22:         pass
   23:     
   24:     def add_edge(self, source, destination):
   25:         '''
   26:         Add an edge to the graph.
--------------------------------------------------------------------------------
[0.20974 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_init

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad5c7e830>

    def test_init(self):
        graph = Graph(5)
        assert len(graph.graph) == 0
>       assert graph.verticies == 5
E       AttributeError: 'Graph' object has no attribute 'verticies'

ts-3-5/test_3-5_1-0_309.py:33: AttributeError
   - [#  86] SDL graph1: 
--------------------------------------------------------------------------------
   36:         if len(self.graph) > self.verticies:
   37:             raise IndexError('Too many verticies in graph.')
   38:         
   39:         if source in self.graph:
-  40:             self.graph[source].append(destination)
+  40:             pass
   41:         else:
   42:             self.graph[source] = [destination]
   43:         
   44:         if destination not in self.graph:
--------------------------------------------------------------------------------
[0.20972 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_has_cycle

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad562eda0>

    def test_has_cycle(self):
        graph = Graph(3)
        graph.add_edge(0, 1)
        graph.add_edge(1, 2)
        assert not graph.has_cycle()
    
        graph.add_edge(2, 0)
>       assert graph.has_cycle()
E       AssertionError

ts-3-5/test_3-5_1-0_309.py:49: AssertionError
   - [#  87] SDL graph1: 
--------------------------------------------------------------------------------
   38:         
   39:         if source in self.graph:
   40:             self.graph[source].append(destination)
   41:         else:
-  42:             self.graph[source] = [destination]
+  42:             pass
   43:         
   44:         if destination not in self.graph:
   45:             self.graph[destination] = []
   46:     
--------------------------------------------------------------------------------
[0.20608 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_add_edge

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad5b4d570>

    def test_add_edge(self):
        graph = Graph(5)
        graph.add_edge(1, 2)
>       assert len(graph.graph) == 2
E       AssertionError

ts-3-5/test_3-5_1-0_309.py:38: AssertionError
   - [#  88] SDL graph1: 
--------------------------------------------------------------------------------
   41:         else:
   42:             self.graph[source] = [destination]
   43:         
   44:         if destination not in self.graph:
-  45:             self.graph[destination] = []
+  45:             pass
   46:     
   47:     def has_cycle(self):
   48:         '''
   49:         Detect if a graph has a cycle.
--------------------------------------------------------------------------------
[0.20643 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_add_edge

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad5b9f3d0>

    def test_add_edge(self):
        graph = Graph(5)
        graph.add_edge(1, 2)
>       assert len(graph.graph) == 2
E       AssertionError

ts-3-5/test_3-5_1-0_309.py:38: AssertionError
   - [#  89] SDL graph1: 
--------------------------------------------------------------------------------
   53:             False if the graph is acyclic.
   54: 
   55:         '''
   56:         
-  57:         visited = [0] * self.verticies
+  57:         pass
   58:         
   59:         def valid(node):
   60:             
   61:             if visited[node] == -1:
--------------------------------------------------------------------------------
[0.21161 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_has_cycle

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad5ab72e0>

    def test_has_cycle(self):
        graph = Graph(3)
        graph.add_edge(0, 1)
        graph.add_edge(1, 2)
>       assert not graph.has_cycle()

ts-3-5/test_3-5_1-0_309.py:46: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.Graph object at 0x7d3ad5ab4f70>

>   ???
E   UnboundLocalError: local variable 'visited' referenced before assignment

graph1:75: UnboundLocalError
   - [#  90] SDL graph1: 
--------------------------------------------------------------------------------
   58:         
   59:         def valid(node):
   60:             
   61:             if visited[node] == -1:
-  62:                 return False
+  62:                 pass
   63:             elif visited[node] == 1:
   64:                 return True
   65:             visited[node] = -1
   66:             for neighbor in self.graph[node]:
--------------------------------------------------------------------------------
[0.21557 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_has_cycle

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad5495c90>

    def test_has_cycle(self):
        graph = Graph(3)
        graph.add_edge(0, 1)
        graph.add_edge(1, 2)
        assert not graph.has_cycle()
    
        graph.add_edge(2, 0)
>       assert graph.has_cycle()

ts-3-5/test_3-5_1-0_309.py:49: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
graph1:77: in has_cycle
    ???
graph1:67: in valid
    ???
graph1:67: in valid
    ???
graph1:67: in valid
    ???
graph1:67: in valid
    ???
E   RecursionError: maximum recursion depth exceeded in comparison
!!! Recursion detected (same locals & position)
   - [#  91] SDL graph1: 
--------------------------------------------------------------------------------
   60:             
   61:             if visited[node] == -1:
   62:                 return False
   63:             elif visited[node] == 1:
-  64:                 return True
+  64:                 pass
   65:             visited[node] = -1
   66:             for neighbor in self.graph[node]:
   67:                 if not (valid(neighbor)):
   68:                     return False
--------------------------------------------------------------------------------
[0.18618 s] survived
   - [#  92] SDL graph1: 
--------------------------------------------------------------------------------
   61:             if visited[node] == -1:
   62:                 return False
   63:             elif visited[node] == 1:
   64:                 return True
-  65:             visited[node] = -1
+  65:             pass
   66:             for neighbor in self.graph[node]:
   67:                 if not (valid(neighbor)):
   68:                     return False
   69:             return True
--------------------------------------------------------------------------------
[0.21692 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_has_cycle

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad54eaec0>

    def test_has_cycle(self):
        graph = Graph(3)
        graph.add_edge(0, 1)
        graph.add_edge(1, 2)
        assert not graph.has_cycle()
    
        graph.add_edge(2, 0)
>       assert graph.has_cycle()

ts-3-5/test_3-5_1-0_309.py:49: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
graph1:77: in has_cycle
    ???
graph1:67: in valid
    ???
graph1:67: in valid
    ???
graph1:67: in valid
    ???
graph1:67: in valid
    ???
E   RecursionError: maximum recursion depth exceeded in comparison
!!! Recursion detected (same locals & position)
   - [#  93] SDL graph1: 
--------------------------------------------------------------------------------
   64:                 return True
   65:             visited[node] = -1
   66:             for neighbor in self.graph[node]:
   67:                 if not (valid(neighbor)):
-  68:                     return False
+  68:                     pass
   69:             return True
   70:         
   71:         for node in range(self.verticies):
   72:             
--------------------------------------------------------------------------------
[0.20853 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_has_cycle

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad562f280>

    def test_has_cycle(self):
        graph = Graph(3)
        graph.add_edge(0, 1)
        graph.add_edge(1, 2)
        assert not graph.has_cycle()
    
        graph.add_edge(2, 0)
>       assert graph.has_cycle()
E       AssertionError

ts-3-5/test_3-5_1-0_309.py:49: AssertionError
   - [#  94] SDL graph1: 
--------------------------------------------------------------------------------
   65:             visited[node] = -1
   66:             for neighbor in self.graph[node]:
   67:                 if not (valid(neighbor)):
   68:                     return False
-  69:             return True
+  69:             pass
   70:         
   71:         for node in range(self.verticies):
   72:             
   73:             
--------------------------------------------------------------------------------
[0.21209 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_has_cycle

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad560d2a0>

    def test_has_cycle(self):
        graph = Graph(3)
        graph.add_edge(0, 1)
        graph.add_edge(1, 2)
>       assert not graph.has_cycle()
E       AssertionError

ts-3-5/test_3-5_1-0_309.py:46: AssertionError
   - [#  95] SDL graph1: 
--------------------------------------------------------------------------------
   75:             if not (visited[node]):
   76:                 
   77:                 if valid(node):
   78:                     
-  79:                     visited = list(map((lambda x: (abs(x))), visited))
+  79:                     pass
   80:                 else:
   81:                     
   82:                     return True
   83:         
--------------------------------------------------------------------------------
[0.18682 s] survived
   - [#  96] SDL graph1: 
--------------------------------------------------------------------------------
   78:                     
   79:                     visited = list(map((lambda x: (abs(x))), visited))
   80:                 else:
   81:                     
-  82:                     return True
+  82:                     pass
   83:         
   84:         return False
   85:     
   86:     def topological_sort(self):
--------------------------------------------------------------------------------
[0.23002 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_has_cycle

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad5760f70>

    def test_has_cycle(self):
        graph = Graph(3)
        graph.add_edge(0, 1)
        graph.add_edge(1, 2)
        assert not graph.has_cycle()
    
        graph.add_edge(2, 0)
>       assert graph.has_cycle()
E       AssertionError

ts-3-5/test_3-5_1-0_309.py:49: AssertionError
   - [#  97] SDL graph1: 
--------------------------------------------------------------------------------
   80:                 else:
   81:                     
   82:                     return True
   83:         
-  84:         return False
+  84:         pass
   85:     
   86:     def topological_sort(self):
   87:         """
   88:         Sort the graph topologically.
--------------------------------------------------------------------------------
[0.18788 s] survived
   - [#  98] SDL graph1: 
--------------------------------------------------------------------------------
  103:         Returns:
  104:             A list of vertices in a topological ordering.
  105: 
  106:         """
- 107:         visited = set()
+ 107:         pass
  108:         stack = deque()
  109:         
  110:         def dfs(vertex):
  111:             visited.add(vertex)
--------------------------------------------------------------------------------
[0.21015 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_topological_sort

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad54f1e40>

    def test_topological_sort(self):
        graph = Graph(4)
        graph.add_edge(0, 1)
        graph.add_edge(1, 2)
        graph.add_edge(1, 3)
        graph.add_edge(2, 3)
>       assert graph.topological_sort() == [0, 1, 2, 3]

ts-3-5/test_3-5_1-0_309.py:57: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.Graph object at 0x7d3ad54f29e0>

>   ???
E   NameError: name 'visited' is not defined

graph1:119: NameError
   - [#  99] SDL graph1: 
--------------------------------------------------------------------------------
  104:             A list of vertices in a topological ordering.
  105: 
  106:         """
  107:         visited = set()
- 108:         stack = deque()
+ 108:         pass
  109:         
  110:         def dfs(vertex):
  111:             visited.add(vertex)
  112:             for j in self.graph[vertex]:
--------------------------------------------------------------------------------
[0.23507 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_topological_sort

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad5852800>

    def test_topological_sort(self):
        graph = Graph(4)
        graph.add_edge(0, 1)
        graph.add_edge(1, 2)
        graph.add_edge(1, 3)
        graph.add_edge(2, 3)
>       assert graph.topological_sort() == [0, 1, 2, 3]

ts-3-5/test_3-5_1-0_309.py:57: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
graph1:120: in topological_sort
    ???
graph1:114: in dfs
    ???
graph1:114: in dfs
    ???
graph1:114: in dfs
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

vertex = 3

>   ???
E   NameError: name 'stack' is not defined

graph1:116: NameError
   - [# 100] SDL graph1: 
--------------------------------------------------------------------------------
  107:         visited = set()
  108:         stack = deque()
  109:         
  110:         def dfs(vertex):
- 111:             visited.add(vertex)
+ 111:             pass
  112:             for j in self.graph[vertex]:
  113:                 if j not in visited:
  114:                     dfs(j)
  115:             
--------------------------------------------------------------------------------
[0.20817 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_topological_sort

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad57efbe0>

    def test_topological_sort(self):
        graph = Graph(4)
        graph.add_edge(0, 1)
        graph.add_edge(1, 2)
        graph.add_edge(1, 3)
        graph.add_edge(2, 3)
>       assert graph.topological_sort() == [0, 1, 2, 3]
E       AssertionError

ts-3-5/test_3-5_1-0_309.py:57: AssertionError
   - [# 101] SDL graph1: 
--------------------------------------------------------------------------------
  110:         def dfs(vertex):
  111:             visited.add(vertex)
  112:             for j in self.graph[vertex]:
  113:                 if j not in visited:
- 114:                     dfs(j)
+ 114:                     pass
  115:             
  116:             stack.appendleft(vertex)
  117:         
  118:         for (key, _) in self.graph.items():
--------------------------------------------------------------------------------
[0.21029 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_topological_sort

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad55ff8e0>

    def test_topological_sort(self):
        graph = Graph(4)
        graph.add_edge(0, 1)
        graph.add_edge(1, 2)
        graph.add_edge(1, 3)
        graph.add_edge(2, 3)
>       assert graph.topological_sort() == [0, 1, 2, 3]
E       AssertionError

ts-3-5/test_3-5_1-0_309.py:57: AssertionError
   - [# 102] SDL graph1: 
--------------------------------------------------------------------------------
  112:             for j in self.graph[vertex]:
  113:                 if j not in visited:
  114:                     dfs(j)
  115:             
- 116:             stack.appendleft(vertex)
+ 116:             pass
  117:         
  118:         for (key, _) in self.graph.items():
  119:             if key not in visited:
  120:                 dfs(key)
--------------------------------------------------------------------------------
[0.21694 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_topological_sort

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad57e4f40>

    def test_topological_sort(self):
        graph = Graph(4)
        graph.add_edge(0, 1)
        graph.add_edge(1, 2)
        graph.add_edge(1, 3)
        graph.add_edge(2, 3)
>       assert graph.topological_sort() == [0, 1, 2, 3]
E       AssertionError

ts-3-5/test_3-5_1-0_309.py:57: AssertionError
   - [# 103] SDL graph1: 
--------------------------------------------------------------------------------
  116:             stack.appendleft(vertex)
  117:         
  118:         for (key, _) in self.graph.items():
  119:             if key not in visited:
- 120:                 dfs(key)
+ 120:                 pass
  121:         
  122:         return list(stack)
  123: 
  124: class WeightedGraphNode:
--------------------------------------------------------------------------------
[0.21149 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_topological_sort

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad5b4e260>

    def test_topological_sort(self):
        graph = Graph(4)
        graph.add_edge(0, 1)
        graph.add_edge(1, 2)
        graph.add_edge(1, 3)
        graph.add_edge(2, 3)
>       assert graph.topological_sort() == [0, 1, 2, 3]
E       AssertionError

ts-3-5/test_3-5_1-0_309.py:57: AssertionError
   - [# 104] SDL graph1: 
--------------------------------------------------------------------------------
  118:         for (key, _) in self.graph.items():
  119:             if key not in visited:
  120:                 dfs(key)
  121:         
- 122:         return list(stack)
+ 122:         pass
  123: 
  124: class WeightedGraphNode:
  125:     
  126:     def __init__(self, val):
--------------------------------------------------------------------------------
[0.21356 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_topological_sort

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad560d6f0>

    def test_topological_sort(self):
        graph = Graph(4)
        graph.add_edge(0, 1)
        graph.add_edge(1, 2)
        graph.add_edge(1, 3)
        graph.add_edge(2, 3)
>       assert graph.topological_sort() == [0, 1, 2, 3]
E       AssertionError

ts-3-5/test_3-5_1-0_309.py:57: AssertionError
   - [# 105] SDL graph1: 
--------------------------------------------------------------------------------
  123: 
  124: class WeightedGraphNode:
  125:     
  126:     def __init__(self, val):
- 127:         self.val = val
+ 127:         pass
  128:         self.adjacent = {}
  129:     
  130:     def __str__(self):
  131:         return (str(self.val) + ' adjacent: ') + str([x for x in self.adjacent])
--------------------------------------------------------------------------------
[0.20798 s] killed by ts-3-5/test_3-5_1-0_309.py::TestWeightedGraphNode::test_init

self = <test_3-5_1-0_309.TestWeightedGraphNode object at 0x7d3ad57631f0>

    def test_init(self):
        w_node = WeightedGraphNode(10)
>       assert w_node.val == 10
E       AttributeError: 'WeightedGraphNode' object has no attribute 'val'

ts-3-5/test_3-5_1-0_309.py:65: AttributeError
   - [# 106] SDL graph1: 
--------------------------------------------------------------------------------
  124: class WeightedGraphNode:
  125:     
  126:     def __init__(self, val):
  127:         self.val = val
- 128:         self.adjacent = {}
+ 128:         pass
  129:     
  130:     def __str__(self):
  131:         return (str(self.val) + ' adjacent: ') + str([x for x in self.adjacent])
  132:     
--------------------------------------------------------------------------------
[0.20882 s] killed by ts-3-5/test_3-5_1-0_309.py::TestWeightedGraphNode::test_init

self = <test_3-5_1-0_309.TestWeightedGraphNode object at 0x7d3ad54c21a0>

    def test_init(self):
        w_node = WeightedGraphNode(10)
        assert w_node.val == 10
>       assert len(w_node.adjacent) == 0
E       AttributeError: 'WeightedGraphNode' object has no attribute 'adjacent'

ts-3-5/test_3-5_1-0_309.py:66: AttributeError
   - [# 107] SDL graph1: 
--------------------------------------------------------------------------------
  127:         self.val = val
  128:         self.adjacent = {}
  129:     
  130:     def __str__(self):
- 131:         return (str(self.val) + ' adjacent: ') + str([x for x in self.adjacent])
+ 131:         pass
  132:     
  133:     def add_adjacent(self, node, weight=0):
  134:         self.adjacent[node] = weight
  135:     
--------------------------------------------------------------------------------
[0.18610 s] survived
   - [# 108] SDL graph1: 
--------------------------------------------------------------------------------
  130:     def __str__(self):
  131:         return (str(self.val) + ' adjacent: ') + str([x for x in self.adjacent])
  132:     
  133:     def add_adjacent(self, node, weight=0):
- 134:         self.adjacent[node] = weight
+ 134:         pass
  135:     
  136:     def remove_adjacent(self, node):
  137:         del self.adjacent[node]
  138: 
--------------------------------------------------------------------------------
[0.20908 s] killed by ts-3-5/test_3-5_1-0_309.py::TestWeightedGraphNode::test_add_adjacent

self = <test_3-5_1-0_309.TestWeightedGraphNode object at 0x7d3ad55df730>

    def test_add_adjacent(self):
        w_node1 = WeightedGraphNode(10)
        w_node2 = WeightedGraphNode(20)
        w_node1.add_adjacent(w_node2, 5)
>       assert len(w_node1.adjacent) == 1
E       AssertionError

ts-3-5/test_3-5_1-0_309.py:72: AssertionError
   - [# 109] SDL graph1: 
--------------------------------------------------------------------------------
  138: 
  139: class WeightedGraph:
  140:     
  141:     def __init__(self):
- 142:         self.vertices = {}
+ 142:         pass
  143:     
  144:     def __str__(self):
  145:         return ', '.join([node.__str__() for node in self.vertices.values()])
  146:     
--------------------------------------------------------------------------------
[0.21278 s] killed by ts-3-5/test_3-5_1-0_309.py::TestWeightedGraph::test_init

self = <test_3-5_1-0_309.TestWeightedGraph object at 0x7d3ad55fee60>

    def test_init(self):
        w_graph = WeightedGraph()
>       assert len(w_graph.vertices) == 0
E       AttributeError: 'WeightedGraph' object has no attribute 'vertices'

ts-3-5/test_3-5_1-0_309.py:88: AttributeError
   - [# 110] SDL graph1: 
--------------------------------------------------------------------------------
  141:     def __init__(self):
  142:         self.vertices = {}
  143:     
  144:     def __str__(self):
- 145:         return ', '.join([node.__str__() for node in self.vertices.values()])
+ 145:         pass
  146:     
  147:     def add_vertex(self, val):
  148:         vertex = WeightedGraphNode(val)
  149:         self.vertices[val] = vertex
--------------------------------------------------------------------------------
[0.18454 s] survived
   - [# 111] SDL graph1: 
--------------------------------------------------------------------------------
  144:     def __str__(self):
  145:         return ', '.join([node.__str__() for node in self.vertices.values()])
  146:     
  147:     def add_vertex(self, val):
- 148:         vertex = WeightedGraphNode(val)
+ 148:         pass
  149:         self.vertices[val] = vertex
  150:     
  151:     def add_edge(self, src, dst, weight=0):
  152:         if src not in self.vertices:
--------------------------------------------------------------------------------
[0.21242 s] killed by ts-3-5/test_3-5_1-0_309.py::TestWeightedGraph::test_add_edge

self = <test_3-5_1-0_309.TestWeightedGraph object at 0x7d3ad571b130>

    def test_add_edge(self):
        w_graph = WeightedGraph()
>       w_graph.add_edge(1, 2, 5)

ts-3-5/test_3-5_1-0_309.py:92: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
graph1:153: in add_edge
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.WeightedGraph object at 0x7d3ad571b940>, val = 1

>   ???
E   NameError: name 'vertex' is not defined

graph1:149: NameError
   - [# 112] SDL graph1: 
--------------------------------------------------------------------------------
  145:         return ', '.join([node.__str__() for node in self.vertices.values()])
  146:     
  147:     def add_vertex(self, val):
  148:         vertex = WeightedGraphNode(val)
- 149:         self.vertices[val] = vertex
+ 149:         pass
  150:     
  151:     def add_edge(self, src, dst, weight=0):
  152:         if src not in self.vertices:
  153:             self.add_vertex(src)
--------------------------------------------------------------------------------
[0.21127 s] killed by ts-3-5/test_3-5_1-0_309.py::TestWeightedGraph::test_add_edge

self = <test_3-5_1-0_309.TestWeightedGraph object at 0x7d3ad554ceb0>

    def test_add_edge(self):
        w_graph = WeightedGraph()
>       w_graph.add_edge(1, 2, 5)

ts-3-5/test_3-5_1-0_309.py:92: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.WeightedGraph object at 0x7d3ad554d660>, src = 1, dst = 2
weight = 5

>   ???
E   KeyError: 1

graph1:156: KeyError
   - [# 113] SDL graph1: 
--------------------------------------------------------------------------------
  149:         self.vertices[val] = vertex
  150:     
  151:     def add_edge(self, src, dst, weight=0):
  152:         if src not in self.vertices:
- 153:             self.add_vertex(src)
+ 153:             pass
  154:         if dst not in self.vertices:
  155:             self.add_vertex(dst)
  156:         self.vertices[src].add_adjacent(dst, weight)
  157:     
--------------------------------------------------------------------------------
[0.21171 s] killed by ts-3-5/test_3-5_1-0_309.py::TestWeightedGraph::test_add_edge

self = <test_3-5_1-0_309.TestWeightedGraph object at 0x7d3ad57e7700>

    def test_add_edge(self):
        w_graph = WeightedGraph()
>       w_graph.add_edge(1, 2, 5)

ts-3-5/test_3-5_1-0_309.py:92: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.WeightedGraph object at 0x7d3ad57e4880>, src = 1, dst = 2
weight = 5

>   ???
E   KeyError: 1

graph1:156: KeyError
   - [# 114] SDL graph1: 
--------------------------------------------------------------------------------
  151:     def add_edge(self, src, dst, weight=0):
  152:         if src not in self.vertices:
  153:             self.add_vertex(src)
  154:         if dst not in self.vertices:
- 155:             self.add_vertex(dst)
+ 155:             pass
  156:         self.vertices[src].add_adjacent(dst, weight)
  157:     
  158:     def remove_edge(self, src, dst):
  159:         if src not in self.vertices:
--------------------------------------------------------------------------------
[0.23158 s] killed by ts-3-5/test_3-5_1-0_309.py::TestWeightedGraph::test_add_edge

self = <test_3-5_1-0_309.TestWeightedGraph object at 0x7d3ad557a7d0>

    def test_add_edge(self):
        w_graph = WeightedGraph()
        w_graph.add_edge(1, 2, 5)
>       assert len(w_graph.vertices) == 2
E       AssertionError

ts-3-5/test_3-5_1-0_309.py:93: AssertionError
   - [# 115] SDL graph1: 
--------------------------------------------------------------------------------
  152:         if src not in self.vertices:
  153:             self.add_vertex(src)
  154:         if dst not in self.vertices:
  155:             self.add_vertex(dst)
- 156:         self.vertices[src].add_adjacent(dst, weight)
+ 156:         pass
  157:     
  158:     def remove_edge(self, src, dst):
  159:         if src not in self.vertices:
  160:             return
--------------------------------------------------------------------------------
[0.23279 s] killed by ts-3-5/test_3-5_1-0_309.py::TestWeightedGraph::test_add_edge

self = <test_3-5_1-0_309.TestWeightedGraph object at 0x7d3ad5852f50>

    def test_add_edge(self):
        w_graph = WeightedGraph()
        w_graph.add_edge(1, 2, 5)
        assert len(w_graph.vertices) == 2
>       assert 2 in w_graph.vertices[1].adjacent
E       AssertionError

ts-3-5/test_3-5_1-0_309.py:94: AssertionError
   - [# 116] SDL graph1: 
--------------------------------------------------------------------------------
  156:         self.vertices[src].add_adjacent(dst, weight)
  157:     
  158:     def remove_edge(self, src, dst):
  159:         if src not in self.vertices:
- 160:             return
+ 160:             pass
  161:         if dst not in self.vertices:
  162:             return
  163:         if dst in self.vertices[src].adjacent:
  164:             self.vertices[src].remove_adjacent(dst)
--------------------------------------------------------------------------------
[0.18614 s] survived
   - [# 117] SDL graph1: 
--------------------------------------------------------------------------------
  158:     def remove_edge(self, src, dst):
  159:         if src not in self.vertices:
  160:             return
  161:         if dst not in self.vertices:
- 162:             return
+ 162:             pass
  163:         if dst in self.vertices[src].adjacent:
  164:             self.vertices[src].remove_adjacent(dst)
--------------------------------------------------------------------------------
[0.18463 s] survived
   - [# 118] SDL graph1: 
--------------------------------------------------------------------------------
  160:             return
  161:         if dst not in self.vertices:
  162:             return
  163:         if dst in self.vertices[src].adjacent:
- 164:             self.vertices[src].remove_adjacent(dst)
+ 164:             pass
--------------------------------------------------------------------------------
[0.21302 s] killed by ts-3-5/test_3-5_1-0_309.py::TestWeightedGraph::test_remove_edge

self = <test_3-5_1-0_309.TestWeightedGraph object at 0x7d3ad55d23e0>

    def test_remove_edge(self):
        w_graph = WeightedGraph()
        w_graph.add_edge(1, 2, 5)
        w_graph.remove_edge(1, 2)
>       assert 2 not in w_graph.vertices[1].adjacent
E       AssertionError

ts-3-5/test_3-5_1-0_309.py:101: AssertionError
   - [# 119] SVD graph1: 
--------------------------------------------------------------------------------
    2: 
    3: class GraphNode:
    4:     
    5:     def __init__(self, val):
-   6:         self.val = val
+   6:         val = val
    7:         self.adjacent_list = set()
    8:     
    9:     def add_adjacent(self, node):
   10:         self.adjacent_list.add(node)
--------------------------------------------------------------------------------
[0.00000 s] incompetent

ValueError: expression must have Store context but has Load instead

   - [# 120] SVD graph1: 
--------------------------------------------------------------------------------
    3: class GraphNode:
    4:     
    5:     def __init__(self, val):
    6:         self.val = val
-   7:         self.adjacent_list = set()
+   7:         adjacent_list = set()
    8:     
    9:     def add_adjacent(self, node):
   10:         self.adjacent_list.add(node)
   11:     
--------------------------------------------------------------------------------
[0.00000 s] incompetent

ValueError: expression must have Store context but has Load instead

   - [# 121] SVD graph1: 
--------------------------------------------------------------------------------
    6:         self.val = val
    7:         self.adjacent_list = set()
    8:     
    9:     def add_adjacent(self, node):
-  10:         self.adjacent_list.add(node)
+  10:         adjacent_list.add(node)
   11:     
   12:     def remove_adjacent(self, node):
   13:         self.adjacent_list.remove(node)
   14: 
--------------------------------------------------------------------------------
[0.20805 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraphNode::test_add_adjacent

self = <test_3-5_1-0_309.TestGraphNode object at 0x7d3ad562f3a0>

    def test_add_adjacent(self):
        node1 = GraphNode(1)
        node2 = GraphNode(2)
>       node1.add_adjacent(node2)

ts-3-5/test_3-5_1-0_309.py:23: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.GraphNode object at 0x7d3ad562e860>
node = <graph1.GraphNode object at 0x7d3ad562fdc0>

>   ???
E   NameError: name 'adjacent_list' is not defined

graph1:10: NameError
   - [# 122] SVD graph1: 
--------------------------------------------------------------------------------
    9:     def add_adjacent(self, node):
   10:         self.adjacent_list.add(node)
   11:     
   12:     def remove_adjacent(self, node):
-  13:         self.adjacent_list.remove(node)
+  13:         adjacent_list.remove(node)
   14: 
   15: class Graph:
   16:     '''
   17:     A directed graph represented with an adjacency list.
--------------------------------------------------------------------------------
[0.18844 s] survived
   - [# 123] SVD graph1: 
--------------------------------------------------------------------------------
   17:     A directed graph represented with an adjacency list.
   18:     '''
   19:     
   20:     def __init__(self, verticies):
-  21:         self.graph = {}
+  21:         graph = {}
   22:         self.verticies = verticies
   23:     
   24:     def add_edge(self, source, destination):
   25:         '''
--------------------------------------------------------------------------------
[0.00000 s] incompetent

ValueError: expression must have Store context but has Load instead

   - [# 124] SVD graph1: 
--------------------------------------------------------------------------------
   18:     '''
   19:     
   20:     def __init__(self, verticies):
   21:         self.graph = {}
-  22:         self.verticies = verticies
+  22:         verticies = verticies
   23:     
   24:     def add_edge(self, source, destination):
   25:         '''
   26:         Add an edge to the graph.
--------------------------------------------------------------------------------
[0.00000 s] incompetent

ValueError: expression must have Store context but has Load instead

   - [# 125] SVD graph1: 
--------------------------------------------------------------------------------
   32:             source: the source vertex
   33:             destination: the destination vertex
   34: 
   35:         '''
-  36:         if len(self.graph) > self.verticies:
+  36:         if len(graph) > self.verticies:
   37:             raise IndexError('Too many verticies in graph.')
   38:         
   39:         if source in self.graph:
   40:             self.graph[source].append(destination)
--------------------------------------------------------------------------------
[0.20445 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_add_edge

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad5c0f850>

    def test_add_edge(self):
        graph = Graph(5)
>       graph.add_edge(1, 2)

ts-3-5/test_3-5_1-0_309.py:37: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.Graph object at 0x7d3ad55fe740>, source = 1, destination = 2

>   ???
E   NameError: name 'graph' is not defined

graph1:36: NameError
   - [# 126] SVD graph1: 
--------------------------------------------------------------------------------
   32:             source: the source vertex
   33:             destination: the destination vertex
   34: 
   35:         '''
-  36:         if len(self.graph) > self.verticies:
+  36:         if len(self.graph) > verticies:
   37:             raise IndexError('Too many verticies in graph.')
   38:         
   39:         if source in self.graph:
   40:             self.graph[source].append(destination)
--------------------------------------------------------------------------------
[0.20562 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_add_edge

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad562eb60>

    def test_add_edge(self):
        graph = Graph(5)
>       graph.add_edge(1, 2)

ts-3-5/test_3-5_1-0_309.py:37: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.Graph object at 0x7d3ad562c850>, source = 1, destination = 2

>   ???
E   NameError: name 'verticies' is not defined

graph1:36: NameError
   - [# 127] SVD graph1: 
--------------------------------------------------------------------------------
   35:         '''
   36:         if len(self.graph) > self.verticies:
   37:             raise IndexError('Too many verticies in graph.')
   38:         
-  39:         if source in self.graph:
+  39:         if source in graph:
   40:             self.graph[source].append(destination)
   41:         else:
   42:             self.graph[source] = [destination]
   43:         
--------------------------------------------------------------------------------
[0.20587 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_add_edge

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad5773df0>

    def test_add_edge(self):
        graph = Graph(5)
>       graph.add_edge(1, 2)

ts-3-5/test_3-5_1-0_309.py:37: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.Graph object at 0x7d3ad5773a90>, source = 1, destination = 2

>   ???
E   NameError: name 'graph' is not defined

graph1:39: NameError
   - [# 128] SVD graph1: 
--------------------------------------------------------------------------------
   36:         if len(self.graph) > self.verticies:
   37:             raise IndexError('Too many verticies in graph.')
   38:         
   39:         if source in self.graph:
-  40:             self.graph[source].append(destination)
+  40:             graph[source].append(destination)
   41:         else:
   42:             self.graph[source] = [destination]
   43:         
   44:         if destination not in self.graph:
--------------------------------------------------------------------------------
[0.21152 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_has_cycle

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad562c0a0>

    def test_has_cycle(self):
        graph = Graph(3)
        graph.add_edge(0, 1)
>       graph.add_edge(1, 2)

ts-3-5/test_3-5_1-0_309.py:45: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.Graph object at 0x7d3ad562d2a0>, source = 1, destination = 2

>   ???
E   NameError: name 'graph' is not defined

graph1:40: NameError
   - [# 129] SVD graph1: 
--------------------------------------------------------------------------------
   38:         
   39:         if source in self.graph:
   40:             self.graph[source].append(destination)
   41:         else:
-  42:             self.graph[source] = [destination]
+  42:             graph[source] = [destination]
   43:         
   44:         if destination not in self.graph:
   45:             self.graph[destination] = []
   46:     
--------------------------------------------------------------------------------
[0.20509 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_add_edge

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad5852f20>

    def test_add_edge(self):
        graph = Graph(5)
>       graph.add_edge(1, 2)

ts-3-5/test_3-5_1-0_309.py:37: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.Graph object at 0x7d3ad5852fb0>, source = 1, destination = 2

>   ???
E   NameError: name 'graph' is not defined

graph1:42: NameError
   - [# 130] SVD graph1: 
--------------------------------------------------------------------------------
   40:             self.graph[source].append(destination)
   41:         else:
   42:             self.graph[source] = [destination]
   43:         
-  44:         if destination not in self.graph:
+  44:         if destination not in graph:
   45:             self.graph[destination] = []
   46:     
   47:     def has_cycle(self):
   48:         '''
--------------------------------------------------------------------------------
[0.20575 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_add_edge

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad552b130>

    def test_add_edge(self):
        graph = Graph(5)
>       graph.add_edge(1, 2)

ts-3-5/test_3-5_1-0_309.py:37: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.Graph object at 0x7d3ad552ba30>, source = 1, destination = 2

>   ???
E   NameError: name 'graph' is not defined

graph1:44: NameError
   - [# 131] SVD graph1: 
--------------------------------------------------------------------------------
   41:         else:
   42:             self.graph[source] = [destination]
   43:         
   44:         if destination not in self.graph:
-  45:             self.graph[destination] = []
+  45:             graph[destination] = []
   46:     
   47:     def has_cycle(self):
   48:         '''
   49:         Detect if a graph has a cycle.
--------------------------------------------------------------------------------
[0.22767 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_add_edge

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad56ce290>

    def test_add_edge(self):
        graph = Graph(5)
>       graph.add_edge(1, 2)

ts-3-5/test_3-5_1-0_309.py:37: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.Graph object at 0x7d3ad565e920>, source = 1, destination = 2

>   ???
E   NameError: name 'graph' is not defined

graph1:45: NameError
   - [# 132] SVD graph1: 
--------------------------------------------------------------------------------
   53:             False if the graph is acyclic.
   54: 
   55:         '''
   56:         
-  57:         visited = [0] * self.verticies
+  57:         visited = [0] * verticies
   58:         
   59:         def valid(node):
   60:             
   61:             if visited[node] == -1:
--------------------------------------------------------------------------------
[0.20934 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_has_cycle

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad5ab51b0>

    def test_has_cycle(self):
        graph = Graph(3)
        graph.add_edge(0, 1)
        graph.add_edge(1, 2)
>       assert not graph.has_cycle()

ts-3-5/test_3-5_1-0_309.py:46: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.Graph object at 0x7d3ad5ab76d0>

>   ???
E   NameError: name 'verticies' is not defined

graph1:57: NameError
   - [# 133] SVD graph1: 
--------------------------------------------------------------------------------
   62:                 return False
   63:             elif visited[node] == 1:
   64:                 return True
   65:             visited[node] = -1
-  66:             for neighbor in self.graph[node]:
+  66:             for neighbor in graph[node]:
   67:                 if not (valid(neighbor)):
   68:                     return False
   69:             return True
   70:         
--------------------------------------------------------------------------------
[0.21026 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_has_cycle

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad55d1060>

    def test_has_cycle(self):
        graph = Graph(3)
        graph.add_edge(0, 1)
        graph.add_edge(1, 2)
>       assert not graph.has_cycle()

ts-3-5/test_3-5_1-0_309.py:46: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
graph1:77: in has_cycle
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

node = 0

>   ???
E   NameError: name 'graph' is not defined

graph1:66: NameError
   - [# 134] SVD graph1: 
--------------------------------------------------------------------------------
   67:                 if not (valid(neighbor)):
   68:                     return False
   69:             return True
   70:         
-  71:         for node in range(self.verticies):
+  71:         for node in range(verticies):
   72:             
   73:             
   74:             
   75:             if not (visited[node]):
--------------------------------------------------------------------------------
[0.20796 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_has_cycle

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad5ab4910>

    def test_has_cycle(self):
        graph = Graph(3)
        graph.add_edge(0, 1)
        graph.add_edge(1, 2)
>       assert not graph.has_cycle()

ts-3-5/test_3-5_1-0_309.py:46: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.Graph object at 0x7d3ad5ab7eb0>

>   ???
E   NameError: name 'verticies' is not defined

graph1:71: NameError
   - [# 135] SVD graph1: 
--------------------------------------------------------------------------------
  108:         stack = deque()
  109:         
  110:         def dfs(vertex):
  111:             visited.add(vertex)
- 112:             for j in self.graph[vertex]:
+ 112:             for j in graph[vertex]:
  113:                 if j not in visited:
  114:                     dfs(j)
  115:             
  116:             stack.appendleft(vertex)
--------------------------------------------------------------------------------
[0.22718 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_topological_sort

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad5793e80>

    def test_topological_sort(self):
        graph = Graph(4)
        graph.add_edge(0, 1)
        graph.add_edge(1, 2)
        graph.add_edge(1, 3)
        graph.add_edge(2, 3)
>       assert graph.topological_sort() == [0, 1, 2, 3]

ts-3-5/test_3-5_1-0_309.py:57: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
graph1:120: in topological_sort
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

vertex = 0

>   ???
E   NameError: name 'graph' is not defined

graph1:112: NameError
   - [# 136] SVD graph1: 
--------------------------------------------------------------------------------
  114:                     dfs(j)
  115:             
  116:             stack.appendleft(vertex)
  117:         
- 118:         for (key, _) in self.graph.items():
+ 118:         for (key, _) in graph.items():
  119:             if key not in visited:
  120:                 dfs(key)
  121:         
  122:         return list(stack)
--------------------------------------------------------------------------------
[0.20482 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_topological_sort

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad5611660>

    def test_topological_sort(self):
        graph = Graph(4)
        graph.add_edge(0, 1)
        graph.add_edge(1, 2)
        graph.add_edge(1, 3)
        graph.add_edge(2, 3)
>       assert graph.topological_sort() == [0, 1, 2, 3]

ts-3-5/test_3-5_1-0_309.py:57: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.Graph object at 0x7d3ad565f550>

>   ???
E   NameError: name 'graph' is not defined

graph1:118: NameError
   - [# 137] SVD graph1: 
--------------------------------------------------------------------------------
  123: 
  124: class WeightedGraphNode:
  125:     
  126:     def __init__(self, val):
- 127:         self.val = val
+ 127:         val = val
  128:         self.adjacent = {}
  129:     
  130:     def __str__(self):
  131:         return (str(self.val) + ' adjacent: ') + str([x for x in self.adjacent])
--------------------------------------------------------------------------------
[0.00000 s] incompetent

ValueError: expression must have Store context but has Load instead

   - [# 138] SVD graph1: 
--------------------------------------------------------------------------------
  124: class WeightedGraphNode:
  125:     
  126:     def __init__(self, val):
  127:         self.val = val
- 128:         self.adjacent = {}
+ 128:         adjacent = {}
  129:     
  130:     def __str__(self):
  131:         return (str(self.val) + ' adjacent: ') + str([x for x in self.adjacent])
  132:     
--------------------------------------------------------------------------------
[0.00000 s] incompetent

ValueError: expression must have Store context but has Load instead

   - [# 139] SVD graph1: 
--------------------------------------------------------------------------------
  127:         self.val = val
  128:         self.adjacent = {}
  129:     
  130:     def __str__(self):
- 131:         return (str(self.val) + ' adjacent: ') + str([x for x in self.adjacent])
+ 131:         return (str(val) + ' adjacent: ') + str([x for x in self.adjacent])
  132:     
  133:     def add_adjacent(self, node, weight=0):
  134:         self.adjacent[node] = weight
  135:     
--------------------------------------------------------------------------------
[0.18500 s] survived
   - [# 140] SVD graph1: 
--------------------------------------------------------------------------------
  127:         self.val = val
  128:         self.adjacent = {}
  129:     
  130:     def __str__(self):
- 131:         return (str(self.val) + ' adjacent: ') + str([x for x in self.adjacent])
+ 131:         return (str(self.val) + ' adjacent: ') + str([x for x in adjacent])
  132:     
  133:     def add_adjacent(self, node, weight=0):
  134:         self.adjacent[node] = weight
  135:     
--------------------------------------------------------------------------------
[0.19023 s] survived
   - [# 141] SVD graph1: 
--------------------------------------------------------------------------------
  130:     def __str__(self):
  131:         return (str(self.val) + ' adjacent: ') + str([x for x in self.adjacent])
  132:     
  133:     def add_adjacent(self, node, weight=0):
- 134:         self.adjacent[node] = weight
+ 134:         adjacent[node] = weight
  135:     
  136:     def remove_adjacent(self, node):
  137:         del self.adjacent[node]
  138: 
--------------------------------------------------------------------------------
[0.18516 s] killed by ts-3-5/test_3-5_1-0_309.py::TestWeightedGraphNode::test_add_adjacent

self = <test_3-5_1-0_309.TestWeightedGraphNode object at 0x7d3ad5ab4d90>

    def test_add_adjacent(self):
        w_node1 = WeightedGraphNode(10)
        w_node2 = WeightedGraphNode(20)
>       w_node1.add_adjacent(w_node2, 5)

ts-3-5/test_3-5_1-0_309.py:71: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.WeightedGraphNode object at 0x7d3ad5ab5db0>
node = <graph1.WeightedGraphNode object at 0x7d3ad5ab6b90>, weight = 5

>   ???
E   NameError: name 'adjacent' is not defined

graph1:134: NameError
   - [# 142] SVD graph1: 
--------------------------------------------------------------------------------
  133:     def add_adjacent(self, node, weight=0):
  134:         self.adjacent[node] = weight
  135:     
  136:     def remove_adjacent(self, node):
- 137:         del self.adjacent[node]
+ 137:         del adjacent[node]
  138: 
  139: class WeightedGraph:
  140:     
  141:     def __init__(self):
--------------------------------------------------------------------------------
[0.20864 s] killed by ts-3-5/test_3-5_1-0_309.py::TestWeightedGraphNode::test_remove_adjacent

self = <test_3-5_1-0_309.TestWeightedGraphNode object at 0x7d3ad562e710>

    def test_remove_adjacent(self):
        w_node1 = WeightedGraphNode(10)
        w_node2 = WeightedGraphNode(20)
        w_node1.add_adjacent(w_node2, 5)
>       w_node1.remove_adjacent(w_node2)

ts-3-5/test_3-5_1-0_309.py:79: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.WeightedGraphNode object at 0x7d3ad5ab6da0>
node = <graph1.WeightedGraphNode object at 0x7d3ad5ab7940>

>   ???
E   NameError: name 'adjacent' is not defined

graph1:137: NameError
   - [# 143] SVD graph1: 
--------------------------------------------------------------------------------
  138: 
  139: class WeightedGraph:
  140:     
  141:     def __init__(self):
- 142:         self.vertices = {}
+ 142:         vertices = {}
  143:     
  144:     def __str__(self):
  145:         return ', '.join([node.__str__() for node in self.vertices.values()])
  146:     
--------------------------------------------------------------------------------
[0.00000 s] incompetent

ValueError: expression must have Store context but has Load instead

   - [# 144] SVD graph1: 
--------------------------------------------------------------------------------
  141:     def __init__(self):
  142:         self.vertices = {}
  143:     
  144:     def __str__(self):
- 145:         return ', '.join([node.__str__() for node in self.vertices.values()])
+ 145:         return ', '.join([node.__str__() for node in vertices.values()])
  146:     
  147:     def add_vertex(self, val):
  148:         vertex = WeightedGraphNode(val)
  149:         self.vertices[val] = vertex
--------------------------------------------------------------------------------
[0.18436 s] survived
   - [# 145] SVD graph1: 
--------------------------------------------------------------------------------
  145:         return ', '.join([node.__str__() for node in self.vertices.values()])
  146:     
  147:     def add_vertex(self, val):
  148:         vertex = WeightedGraphNode(val)
- 149:         self.vertices[val] = vertex
+ 149:         vertices[val] = vertex
  150:     
  151:     def add_edge(self, src, dst, weight=0):
  152:         if src not in self.vertices:
  153:             self.add_vertex(src)
--------------------------------------------------------------------------------
[0.20627 s] killed by ts-3-5/test_3-5_1-0_309.py::TestWeightedGraph::test_add_edge

self = <test_3-5_1-0_309.TestWeightedGraph object at 0x7d3ad56ccac0>

    def test_add_edge(self):
        w_graph = WeightedGraph()
>       w_graph.add_edge(1, 2, 5)

ts-3-5/test_3-5_1-0_309.py:92: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
graph1:153: in add_edge
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.WeightedGraph object at 0x7d3ad56cded0>, val = 1

>   ???
E   NameError: name 'vertices' is not defined

graph1:149: NameError
   - [# 146] SVD graph1: 
--------------------------------------------------------------------------------
  148:         vertex = WeightedGraphNode(val)
  149:         self.vertices[val] = vertex
  150:     
  151:     def add_edge(self, src, dst, weight=0):
- 152:         if src not in self.vertices:
+ 152:         if src not in vertices:
  153:             self.add_vertex(src)
  154:         if dst not in self.vertices:
  155:             self.add_vertex(dst)
  156:         self.vertices[src].add_adjacent(dst, weight)
--------------------------------------------------------------------------------
[0.21305 s] killed by ts-3-5/test_3-5_1-0_309.py::TestWeightedGraph::test_add_edge

self = <test_3-5_1-0_309.TestWeightedGraph object at 0x7d3ad54ce410>

    def test_add_edge(self):
        w_graph = WeightedGraph()
>       w_graph.add_edge(1, 2, 5)

ts-3-5/test_3-5_1-0_309.py:92: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.WeightedGraph object at 0x7d3ad54ce800>, src = 1, dst = 2
weight = 5

>   ???
E   NameError: name 'vertices' is not defined

graph1:152: NameError
   - [# 147] SVD graph1: 
--------------------------------------------------------------------------------
  149:         self.vertices[val] = vertex
  150:     
  151:     def add_edge(self, src, dst, weight=0):
  152:         if src not in self.vertices:
- 153:             self.add_vertex(src)
+ 153:             add_vertex(src)
  154:         if dst not in self.vertices:
  155:             self.add_vertex(dst)
  156:         self.vertices[src].add_adjacent(dst, weight)
  157:     
--------------------------------------------------------------------------------
[0.23379 s] killed by ts-3-5/test_3-5_1-0_309.py::TestWeightedGraph::test_add_edge

self = <test_3-5_1-0_309.TestWeightedGraph object at 0x7d3ad55a8250>

    def test_add_edge(self):
        w_graph = WeightedGraph()
>       w_graph.add_edge(1, 2, 5)

ts-3-5/test_3-5_1-0_309.py:92: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.WeightedGraph object at 0x7d3ad55a8880>, src = 1, dst = 2
weight = 5

>   ???
E   NameError: name 'add_vertex' is not defined

graph1:153: NameError
   - [# 148] SVD graph1: 
--------------------------------------------------------------------------------
  150:     
  151:     def add_edge(self, src, dst, weight=0):
  152:         if src not in self.vertices:
  153:             self.add_vertex(src)
- 154:         if dst not in self.vertices:
+ 154:         if dst not in vertices:
  155:             self.add_vertex(dst)
  156:         self.vertices[src].add_adjacent(dst, weight)
  157:     
  158:     def remove_edge(self, src, dst):
--------------------------------------------------------------------------------
[0.21017 s] killed by ts-3-5/test_3-5_1-0_309.py::TestWeightedGraph::test_add_edge

self = <test_3-5_1-0_309.TestWeightedGraph object at 0x7d3ad5a67490>

    def test_add_edge(self):
        w_graph = WeightedGraph()
>       w_graph.add_edge(1, 2, 5)

ts-3-5/test_3-5_1-0_309.py:92: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.WeightedGraph object at 0x7d3ad5a67370>, src = 1, dst = 2
weight = 5

>   ???
E   NameError: name 'vertices' is not defined

graph1:154: NameError
   - [# 149] SVD graph1: 
--------------------------------------------------------------------------------
  151:     def add_edge(self, src, dst, weight=0):
  152:         if src not in self.vertices:
  153:             self.add_vertex(src)
  154:         if dst not in self.vertices:
- 155:             self.add_vertex(dst)
+ 155:             add_vertex(dst)
  156:         self.vertices[src].add_adjacent(dst, weight)
  157:     
  158:     def remove_edge(self, src, dst):
  159:         if src not in self.vertices:
--------------------------------------------------------------------------------
[0.20700 s] killed by ts-3-5/test_3-5_1-0_309.py::TestWeightedGraph::test_add_edge

self = <test_3-5_1-0_309.TestWeightedGraph object at 0x7d3ad55d2b30>

    def test_add_edge(self):
        w_graph = WeightedGraph()
>       w_graph.add_edge(1, 2, 5)

ts-3-5/test_3-5_1-0_309.py:92: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.WeightedGraph object at 0x7d3ad55d3940>, src = 1, dst = 2
weight = 5

>   ???
E   NameError: name 'add_vertex' is not defined

graph1:155: NameError
   - [# 150] SVD graph1: 
--------------------------------------------------------------------------------
  152:         if src not in self.vertices:
  153:             self.add_vertex(src)
  154:         if dst not in self.vertices:
  155:             self.add_vertex(dst)
- 156:         self.vertices[src].add_adjacent(dst, weight)
+ 156:         vertices[src].add_adjacent(dst, weight)
  157:     
  158:     def remove_edge(self, src, dst):
  159:         if src not in self.vertices:
  160:             return
--------------------------------------------------------------------------------
[0.20909 s] killed by ts-3-5/test_3-5_1-0_309.py::TestWeightedGraph::test_add_edge

self = <test_3-5_1-0_309.TestWeightedGraph object at 0x7d3ad55d0970>

    def test_add_edge(self):
        w_graph = WeightedGraph()
>       w_graph.add_edge(1, 2, 5)

ts-3-5/test_3-5_1-0_309.py:92: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.WeightedGraph object at 0x7d3ad55d3550>, src = 1, dst = 2
weight = 5

>   ???
E   NameError: name 'vertices' is not defined

graph1:156: NameError
   - [# 151] SVD graph1: 
--------------------------------------------------------------------------------
  155:             self.add_vertex(dst)
  156:         self.vertices[src].add_adjacent(dst, weight)
  157:     
  158:     def remove_edge(self, src, dst):
- 159:         if src not in self.vertices:
+ 159:         if src not in vertices:
  160:             return
  161:         if dst not in self.vertices:
  162:             return
  163:         if dst in self.vertices[src].adjacent:
--------------------------------------------------------------------------------
[0.21562 s] killed by ts-3-5/test_3-5_1-0_309.py::TestWeightedGraph::test_remove_edge

self = <test_3-5_1-0_309.TestWeightedGraph object at 0x7d3ad574f3a0>

    def test_remove_edge(self):
        w_graph = WeightedGraph()
        w_graph.add_edge(1, 2, 5)
>       w_graph.remove_edge(1, 2)

ts-3-5/test_3-5_1-0_309.py:100: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.WeightedGraph object at 0x7d3ad574c250>, src = 1, dst = 2

>   ???
E   NameError: name 'vertices' is not defined

graph1:159: NameError
   - [# 152] SVD graph1: 
--------------------------------------------------------------------------------
  157:     
  158:     def remove_edge(self, src, dst):
  159:         if src not in self.vertices:
  160:             return
- 161:         if dst not in self.vertices:
+ 161:         if dst not in vertices:
  162:             return
  163:         if dst in self.vertices[src].adjacent:
  164:             self.vertices[src].remove_adjacent(dst)
--------------------------------------------------------------------------------
[0.21089 s] killed by ts-3-5/test_3-5_1-0_309.py::TestWeightedGraph::test_remove_edge

self = <test_3-5_1-0_309.TestWeightedGraph object at 0x7d3ad5c7e680>

    def test_remove_edge(self):
        w_graph = WeightedGraph()
        w_graph.add_edge(1, 2, 5)
>       w_graph.remove_edge(1, 2)

ts-3-5/test_3-5_1-0_309.py:100: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.WeightedGraph object at 0x7d3ad5c7d9c0>, src = 1, dst = 2

>   ???
E   NameError: name 'vertices' is not defined

graph1:161: NameError
   - [# 153] SVD graph1: 
--------------------------------------------------------------------------------
  159:         if src not in self.vertices:
  160:             return
  161:         if dst not in self.vertices:
  162:             return
- 163:         if dst in self.vertices[src].adjacent:
+ 163:         if dst in vertices[src].adjacent:
  164:             self.vertices[src].remove_adjacent(dst)
--------------------------------------------------------------------------------
[0.21108 s] killed by ts-3-5/test_3-5_1-0_309.py::TestWeightedGraph::test_remove_edge

self = <test_3-5_1-0_309.TestWeightedGraph object at 0x7d3ad575a8c0>

    def test_remove_edge(self):
        w_graph = WeightedGraph()
        w_graph.add_edge(1, 2, 5)
>       w_graph.remove_edge(1, 2)

ts-3-5/test_3-5_1-0_309.py:100: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.WeightedGraph object at 0x7d3ad575a410>, src = 1, dst = 2

>   ???
E   NameError: name 'vertices' is not defined

graph1:163: NameError
   - [# 154] SVD graph1: 
--------------------------------------------------------------------------------
  160:             return
  161:         if dst not in self.vertices:
  162:             return
  163:         if dst in self.vertices[src].adjacent:
- 164:             self.vertices[src].remove_adjacent(dst)
+ 164:             vertices[src].remove_adjacent(dst)
--------------------------------------------------------------------------------
[0.21101 s] killed by ts-3-5/test_3-5_1-0_309.py::TestWeightedGraph::test_remove_edge

self = <test_3-5_1-0_309.TestWeightedGraph object at 0x7d3ad54cece0>

    def test_remove_edge(self):
        w_graph = WeightedGraph()
        w_graph.add_edge(1, 2, 5)
>       w_graph.remove_edge(1, 2)

ts-3-5/test_3-5_1-0_309.py:100: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.WeightedGraph object at 0x7d3ad54cef50>, src = 1, dst = 2

>   ???
E   NameError: name 'vertices' is not defined

graph1:164: NameError
   - [# 155] ZIL graph1: 
--------------------------------------------------------------------------------
   63:             elif visited[node] == 1:
   64:                 return True
   65:             visited[node] = -1
   66:             for neighbor in self.graph[node]:
-  67:                 if not (valid(neighbor)):
-  68:                     return False
+  67:                 break
+  68:             
   69:             return True
   70:         
   71:         for node in range(self.verticies):
   72:             
--------------------------------------------------------------------------------
[0.23301 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_has_cycle

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad5529510>

    def test_has_cycle(self):
        graph = Graph(3)
        graph.add_edge(0, 1)
        graph.add_edge(1, 2)
        assert not graph.has_cycle()
    
        graph.add_edge(2, 0)
>       assert graph.has_cycle()
E       AssertionError

ts-3-5/test_3-5_1-0_309.py:49: AssertionError
   - [# 156] ZIL graph1: 
--------------------------------------------------------------------------------
   71:         for node in range(self.verticies):
   72:             
   73:             
   74:             
-  75:             if not (visited[node]):
-  76:                 
-  77:                 if valid(node):
-  78:                     
-  79:                     visited = list(map((lambda x: (abs(x))), visited))
-  80:                 else:
-  81:                     
-  82:                     return True
+  75:             break
+  76:         
+  77:         
+  78:         
+  79:         
+  80:         
+  81:         
+  82:         
   83:         
   84:         return False
   85:     
   86:     def topological_sort(self):
--------------------------------------------------------------------------------
[0.23361 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_has_cycle

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad562c3a0>

    def test_has_cycle(self):
        graph = Graph(3)
        graph.add_edge(0, 1)
        graph.add_edge(1, 2)
        assert not graph.has_cycle()
    
        graph.add_edge(2, 0)
>       assert graph.has_cycle()
E       AssertionError

ts-3-5/test_3-5_1-0_309.py:49: AssertionError
   - [# 157] ZIL graph1: 
--------------------------------------------------------------------------------
  109:         
  110:         def dfs(vertex):
  111:             visited.add(vertex)
  112:             for j in self.graph[vertex]:
- 113:                 if j not in visited:
- 114:                     dfs(j)
+ 113:                 break
+ 114:             
  115:             
  116:             stack.appendleft(vertex)
  117:         
  118:         for (key, _) in self.graph.items():
--------------------------------------------------------------------------------
[0.20874 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_topological_sort

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad5719750>

    def test_topological_sort(self):
        graph = Graph(4)
        graph.add_edge(0, 1)
        graph.add_edge(1, 2)
        graph.add_edge(1, 3)
        graph.add_edge(2, 3)
>       assert graph.topological_sort() == [0, 1, 2, 3]
E       AssertionError

ts-3-5/test_3-5_1-0_309.py:57: AssertionError
   - [# 158] ZIL graph1: 
--------------------------------------------------------------------------------
  115:             
  116:             stack.appendleft(vertex)
  117:         
  118:         for (key, _) in self.graph.items():
- 119:             if key not in visited:
- 120:                 dfs(key)
+ 119:             break
+ 120:         
  121:         
  122:         return list(stack)
  123: 
  124: class WeightedGraphNode:
--------------------------------------------------------------------------------
[0.21150 s] killed by ts-3-5/test_3-5_1-0_309.py::TestGraph::test_topological_sort

self = <test_3-5_1-0_309.TestGraph object at 0x7d3ad5b9e560>

    def test_topological_sort(self):
        graph = Graph(4)
        graph.add_edge(0, 1)
        graph.add_edge(1, 2)
        graph.add_edge(1, 3)
        graph.add_edge(2, 3)
>       assert graph.topological_sort() == [0, 1, 2, 3]
E       AssertionError

ts-3-5/test_3-5_1-0_309.py:57: AssertionError
[*] Mutation score [42.62992 s]: 76.6%
   - all: 158
   - killed: 105 (66.5%)
   - survived: 32 (20.3%)
   - incompetent: 21 (13.3%)
   - timeout: 0 (0.0%)
